diff --git a/plugins/Makefile.am b/plugins/Makefile.am
index ebb79d5..49d0c62 100644
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -75,7 +75,7 @@ plugindir = @plugindir@
 
 common_sources = plugin_common.c plugin_common.h
 
-sasldir = $(prefix)/lib/sasl2
+sasldir = $(prefix)/lib64/sasl2
 sasl_LTLIBRARIES = @SASL_MECHS@
 EXTRA_LTLIBRARIES = libplain.la libanonymous.la libkerberos4.la libcrammd5.la \
 	libgssapiv2.la libdigestmd5.la liblogin.la libsrp.la libotp.la \
diff --git a/plugins/Makefile.in b/plugins/Makefile.in
index cd52645..d97de69 100644
--- a/plugins/Makefile.in
+++ b/plugins/Makefile.in
@@ -263,7 +263,7 @@ noinst_SCRIPTS = makeinit.sh
 
 common_sources = plugin_common.c plugin_common.h
 
-sasldir = $(prefix)/lib/sasl2
+sasldir = $(prefix)/lib64/sasl2
 sasl_LTLIBRARIES = @SASL_MECHS@
 EXTRA_LTLIBRARIES = libplain.la libanonymous.la libkerberos4.la libcrammd5.la \
 	libgssapiv2.la libdigestmd5.la liblogin.la libsrp.la libotp.la \
