diff --git a/plugins/NTMakefile b/plugins/NTMakefile
index 11f008e..f0c0c77 100644
--- a/plugins/NTMakefile
+++ b/plugins/NTMakefile
@@ -41,7 +41,6 @@ PLUGINS=saslANONYMOUS.dll \
 	saslDIGESTMD5.dll \
 	saslLOGIN.dll \
 	$(PLUGINS_EXT) \
-	saslSASLDB.dll
 
 generated_rc=saslANONYMOUS.rc saslPLAIN.rc saslCRAMMD5.rc saslDIGESTMD5.rc saslLOGIN.rc saslNTLM.rc saslGSSAPI.rc saslSRP.rc saslOTP.rc saslSASLDB.rc saslSQLITE.rc saslLDAPDB.rc
 
