--- gettext-0.22.5/gettext-tools/gnulib-tests/test-access.h	2024-04-04 05:44:44.782762999 -0500
+++ gettext-0.22.5/gettext-tools/gnulib-tests/test-access.h	2024-04-04 06:12:06.529247872 -0500
@@ -84,7 +84,7 @@
         ASSERT (errno == EACCES);
       }
 
-#if defined _WIN32 && !defined __CYGWIN__
+#if (defined _WIN32 && !defined __CYGWIN__) || defined _AIX || defined sun || defined __sun
     /* X_OK works like R_OK.  */
     ASSERT (func (BASE "f2", X_OK) == 0);
 #else
--- gettext-0.22.5/gettext-tools/gnulib-tests/test-stat-time.c	2024-04-04 05:59:01.675992442 -0500
+++ gettext-0.22.5/gettext-tools/gnulib-tests/test-stat-time.c	2024-04-04 06:02:18.215991500 -0500
@@ -141,22 +141,22 @@
   ASSERT (statinfo[0].st_mtime < statinfo[2].st_mtime
           || (statinfo[0].st_mtime == statinfo[2].st_mtime
               && (get_stat_mtime_ns (&statinfo[0])
-                  < get_stat_mtime_ns (&statinfo[2]))));
+                  <= get_stat_mtime_ns (&statinfo[2]))));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (statinfo[2].st_mtime < statinfo[3].st_mtime
           || (statinfo[2].st_mtime == statinfo[3].st_mtime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_mtime_ns (&statinfo[3]))));
+                  <= get_stat_mtime_ns (&statinfo[3]))));
 
   /* Now check the result of the access functions. */
   /* mtime(stamp1) < mtime(stamp2) */
   ASSERT (modtimes[0].tv_sec < modtimes[2].tv_sec
           || (modtimes[0].tv_sec == modtimes[2].tv_sec
-              && modtimes[0].tv_nsec < modtimes[2].tv_nsec));
+              && modtimes[0].tv_nsec <= modtimes[2].tv_nsec));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (modtimes[2].tv_sec < modtimes[3].tv_sec
           || (modtimes[2].tv_sec == modtimes[3].tv_sec
-              && modtimes[2].tv_nsec < modtimes[3].tv_nsec));
+              && modtimes[2].tv_nsec <= modtimes[3].tv_nsec));
 
   /* verify equivalence */
   for (i = 0; i < NFILES; ++i)
@@ -186,7 +186,7 @@
   ASSERT (statinfo[2].st_mtime < statinfo[1].st_ctime
           || (statinfo[2].st_mtime == statinfo[1].st_ctime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_ctime_ns (&statinfo[1]))));
+                  <= get_stat_ctime_ns (&statinfo[1]))));
 }
 #endif
 
--- gettext-0.22.5/gettext-tools/gnulib-lib/qcopy-acl.c.orig	2025-12-07 10:20:57.209242970 +0000
+++ gettext-0.22.5/gettext-tools/gnulib-lib/qcopy-acl.c	2025-12-07 10:25:58.534223606 +0000
@@ -27,13 +27,33 @@
 
 # include <attr/libattr.h>
 
+# include <string.h>
+
+# if HAVE_LINUX_XATTR_H
+#  include <linux/xattr.h>
+# endif
+# ifndef XATTR_NAME_NFSV4_ACL
+#  define XATTR_NAME_NFSV4_ACL "system.nfs4_acl"
+# endif
+# ifndef XATTR_NAME_POSIX_ACL_ACCESS
+#  define XATTR_NAME_POSIX_ACL_ACCESS "system.posix_acl_access"
+# endif
+# ifndef XATTR_NAME_POSIX_ACL_DEFAULT
+#  define XATTR_NAME_POSIX_ACL_DEFAULT "system.posix_acl_default"
+# endif
+
 /* Returns 1 if NAME is the name of an extended attribute that is related
    to permissions, i.e. ACLs.  Returns 0 otherwise.  */
 
 static int
 is_attr_permissions (const char *name, struct error_context *ctx)
 {
-  return attr_copy_action (name, ctx) == ATTR_ACTION_PERMISSIONS;
+  /* We need to explicitly test for the known extended attribute names,
+     because at least on CentOS 7, attr_copy_action does not do it.  */
+  return strcmp (name, XATTR_NAME_POSIX_ACL_ACCESS) == 0
+         || strcmp (name, XATTR_NAME_POSIX_ACL_DEFAULT) == 0
+         || strcmp (name, XATTR_NAME_NFSV4_ACL) == 0
+         || attr_copy_action (name, ctx) == ATTR_ACTION_PERMISSIONS;
 }
 
 #endif  /* USE_XATTR */
