--- gnutls-3.8.11/lib/crau/crau.h	2025-10-24 10:10:22.000000000 +0400
+++ gnutls-3.8.11/lib/crau/crau.h	2026-02-16 19:58:41.449481320 +0400
@@ -251,9 +251,10 @@
 # else
 
 #  ifndef CRAU_MAYBE_UNUSED
-#   if defined(__has_c_attribute) && \
-    __has_c_attribute (__maybe_unused__)
-#    define CRAU_MAYBE_UNUSED [[__maybe_unused__]]
+#   if defined(__has_c_attribute)
+#    if __has_c_attribute (__maybe_unused__)
+#     define CRAU_MAYBE_UNUSED [[__maybe_unused__]]
+#    endif
 #   elif defined(__GNUC__)
 #    define CRAU_MAYBE_UNUSED __attribute__((__unused__))
 #   endif
--- gnutls-3.8.11/src/gl/tests/test-stat-time.c	2025-11-20 06:09:15.000000000 +0400
+++ gnutls-3.8.11/src/gl/tests/test-stat-time.c	2026-02-16 20:08:19.989435426 +0400
@@ -141,22 +141,22 @@
   ASSERT (statinfo[0].st_mtime < statinfo[2].st_mtime
           || (statinfo[0].st_mtime == statinfo[2].st_mtime
               && (get_stat_mtime_ns (&statinfo[0])
-                  < get_stat_mtime_ns (&statinfo[2]))));
+                  <= get_stat_mtime_ns (&statinfo[2]))));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (statinfo[2].st_mtime < statinfo[3].st_mtime
           || (statinfo[2].st_mtime == statinfo[3].st_mtime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_mtime_ns (&statinfo[3]))));
+                  <= get_stat_mtime_ns (&statinfo[3]))));
 
   /* Now check the result of the access functions. */
   /* mtime(stamp1) < mtime(stamp2) */
   ASSERT (modtimes[0].tv_sec < modtimes[2].tv_sec
           || (modtimes[0].tv_sec == modtimes[2].tv_sec
-              && modtimes[0].tv_nsec < modtimes[2].tv_nsec));
+              && modtimes[0].tv_nsec <= modtimes[2].tv_nsec));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (modtimes[2].tv_sec < modtimes[3].tv_sec
           || (modtimes[2].tv_sec == modtimes[3].tv_sec
-              && modtimes[2].tv_nsec < modtimes[3].tv_nsec));
+              && modtimes[2].tv_nsec <= modtimes[3].tv_nsec));
 
   /* verify equivalence */
   for (i = 0; i < NFILES; ++i)
@@ -186,7 +186,7 @@
   ASSERT (statinfo[2].st_mtime < statinfo[1].st_ctime
           || (statinfo[2].st_mtime == statinfo[1].st_ctime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_ctime_ns (&statinfo[1]))));
+                  <= get_stat_ctime_ns (&statinfo[1]))));
 }
 #endif
 
--- gnutls-3.8.11/tests/client-fastopen.c	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/client-fastopen.c	2026-02-17 01:22:31.367388037 +0400
@@ -30,7 +30,7 @@
 /* This tests gnutls_transport_set_fastopen() operation.
  */
 
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(_AIX)
 
 int main(void)
 {
--- gnutls-3.8.11/tests/dtls/dtls-stress.c	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/dtls/dtls-stress.c	2026-02-16 21:29:01.156864644 +0400
@@ -1017,7 +1017,7 @@
 	int pid1, pid2;
 	int status2;
 
-	if (socketpair(AF_LOCAL, SOCK_STREAM, 0, fds) < 0) {
+	if (socketpair(AF_UNIX, SOCK_STREAM, 0, fds) < 0) {
 		rperror("socketpair");
 		exit(2);
 	}
--- gnutls-3.8.11/tests/mini-x509-cas.c	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/mini-x509-cas.c	2026-02-16 19:58:41.454565141 +0400
@@ -36,7 +36,7 @@
 /* Tests whether a very long CA list, that spans over multiple records
  * is tolerated */
 
-const char *side;
+const char *side = "NONE";
 
 static void tls_log_func(int level, const char *str)
 {
--- gnutls-3.8.11/tests/scripts/common.sh	2025-10-14 07:38:03.000000000 +0400
+++ gnutls-3.8.11/tests/scripts/common.sh	2026-02-16 21:37:41.862143929 +0400
@@ -45,8 +45,10 @@
 
 	# We can't assume netstat --version for portability reasons
 	if (type netstat) > /dev/null 2>&1; then
-		PFCMD=netstat
-		return
+		if netstat -l > /dev/null 2>&1; then
+			PFCMD=netstat
+			return
+		fi
 	fi
 
 	echo "neither ss nor netstat found" 1>&2
@@ -279,7 +281,11 @@
 	kernel_major=$(echo $kernel_version | cut -d. -f1 2>/dev/null)
 	kernel_minor=$(echo $kernel_version | cut -d. -f2 2>/dev/null)
 
-	if ! [[ "$kernel_major" =~ ^[0-9]+$ ]] || ! [[ "$kernel_minor" =~ ^[0-9]+$ ]]; then
+	if ! echo "$kernel_major" | grep -E '^[0-9]+$' > /dev/null 2>&1; then
+		return 1
+	fi
+
+	if ! echo "$kernel_minor" | grep -E '^[0-9]+$' > /dev/null 2>&1; then
 		return 1
 	fi
 
--- gnutls-3.8.11/tests/slow/cipher-api-test.c	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/slow/cipher-api-test.c	2026-02-17 01:41:50.249397800 +0400
@@ -36,7 +36,7 @@
  * cases may hit assertion failure in Nettle and crash the process.
  */
 
-#if defined(WIN32)
+#if defined(WIN32) || defined(_AIX)
 int main(int argc, char **argv)
 {
 	exit(77);
--- gnutls-3.8.11/tests/slow/test-hash-large.sh	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/slow/test-hash-large.sh	2026-02-17 01:29:26.014418222 +0400
@@ -21,6 +21,14 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>
 #
 
+UNAME=$(uname)
+
+case "$UNAME" in
+AIX)
+	exit 77
+	;;
+esac
+
 PROG=./hash-large${EXEEXT}
 unset RETCODE
 if ! test -z "${VALGRIND}"; then
--- gnutls-3.8.11/tests/tls-force-etm.c	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/tls-force-etm.c	2026-02-17 01:22:53.452214579 +0400
@@ -26,7 +26,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(_AIX)
 
 int main(void)
 {
--- gnutls-3.8.11/tests/utils.h	2025-09-25 10:22:49.000000000 +0400
+++ gnutls-3.8.11/tests/utils.h	2026-02-16 20:04:06.923088320 +0400
@@ -145,20 +145,28 @@
 /* calls fail() if status indicates an error  */
 inline static void _check_wait_status(int status, unsigned sigonly)
 {
-#if defined WEXITSTATUS && defined WIFSIGNALED
-	if (WEXITSTATUS(status) != 0 ||
-	    (WIFSIGNALED(status) && WTERMSIG(status) != SIGTERM)) {
-		if (WIFSIGNALED(status)) {
-			fail("Child died with signal %d\n", WTERMSIG(status));
-		} else {
-			if (!sigonly) {
-				if (WEXITSTATUS(status) == 77)
-					_exit(77);
-				fail("Child died with status %d\n",
-				     WEXITSTATUS(status));
+#if defined(WIFSIGNALED) && defined(WTERMSIG) && defined(WIFEXITED) && defined(WEXITSTATUS)
+	if (WIFSIGNALED(status)) {
+		int sig = WTERMSIG(status);
+		if (sig != SIGTERM) {
+			fail("Child died with signal %d\n", sig);
+		}
+		return;
+	}
+
+	if (WIFEXITED(status)) {
+		int rc = WEXITSTATUS(status);
+		if (rc != 0 && !sigonly) {
+			if (rc == 77) {
+				_exit(77);
 			}
+			fail("Child died with status %d\n", rc);
 		}
+		return;
 	}
+
+	/* Optional: unexpected wait status */
+	fail("Child died with unknown wait status 0x%x\n", status);
 #endif
 }
 
