--- libbsd-0.12.2/test/Makefile.in	2024-03-30 21:39:13.083573303 +0000
+++ libbsd-0.12.2/test/Makefile.in	2024-03-30 21:39:29.171573826 +0000
@@ -737,6 +737,7 @@
 	-include $(top_builddir)/config.h \
 	-DLIBBSD_DISABLE_DEPRECATED \
 	-D__REENTRANT \
+	$(CFLAGS) \
 	# EOL
 
 AM_CPPFLAGS = \
--- libbsd-0.12.2/Makefile.in	2024-04-01 00:12:43.146679388 +0000
+++ libbsd-0.12.2/Makefile.in	2024-04-01 00:13:02.882680029 +0000
@@ -342,7 +342,6 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = \
 	include \
-	man \
 	src \
 	test \
 	# EOL
