diff -ur libevent-2.1.12-stable.orig/evdns.c libevent-2.1.12-stable/evdns.c
--- libevent-2.1.12-stable.orig/evdns.c	2020-07-05 15:01:34.000000000 +0300
+++ libevent-2.1.12-stable/evdns.c	2024-06-01 23:13:07.747352427 +0300
@@ -4035,7 +4035,7 @@
 #else
 		r = evdns_base_resolv_conf_parse(base, opts, "/etc/resolv.conf");
 #endif
-		if (r) {
+		if (r == -1) {
 			evdns_base_free_and_unlock(base, 0);
 			return NULL;
 		}
diff -ur libevent-2.1.12-stable.orig/test/regress_main.c libevent-2.1.12-stable/test/regress_main.c
--- libevent-2.1.12-stable.orig/test/regress_main.c	2020-07-05 15:01:34.000000000 +0300
+++ libevent-2.1.12-stable/test/regress_main.c	2024-06-01 23:21:48.451169707 +0300
@@ -431,8 +431,6 @@
 	{ "signal/", signal_testcases },
 	{ "util/", util_testcases },
 	{ "bufferevent/", bufferevent_testcases },
-	{ "http/", http_testcases },
-	{ "dns/", dns_testcases },
 	{ "evtag/", evtag_testcases },
 	{ "rpc/", rpc_testcases },
 	{ "thread/", thread_testcases },
diff -ur libevent-2.1.12-stable.orig/test/regress_util.c libevent-2.1.12-stable/test/regress_util.c
--- libevent-2.1.12-stable.orig/test/regress_util.c	2020-07-05 15:01:34.000000000 +0300
+++ libevent-2.1.12-stable/test/regress_util.c	2024-06-01 23:22:13.022857465 +0300
@@ -1672,7 +1672,6 @@
 	{ "monotonic_res_fallback", test_evutil_monotonic_res, TT_OFF_BY_DEFAULT, &basic_setup, (void*)"fallback" },
 	{ "monotonic_prc", test_evutil_monotonic_prc, 0, &basic_setup, (void*)"" },
 	{ "monotonic_prc_precise", test_evutil_monotonic_prc, TT_RETRIABLE, &basic_setup, (void*)"precise" },
-	{ "monotonic_prc_fallback", test_evutil_monotonic_prc, 0, &basic_setup, (void*)"fallback" },
 	{ "date_rfc1123", test_evutil_date_rfc1123, 0, NULL, NULL },
 	{ "evutil_v4addr_is_local", test_evutil_v4addr_is_local, 0, NULL, NULL },
 	{ "evutil_v6addr_is_local", test_evutil_v6addr_is_local, 0, NULL, NULL },
diff -ur libevent-2.1.12-stable.orig/test/test.sh libevent-2.1.12-stable/test/test.sh
--- libevent-2.1.12-stable.orig/test/test.sh	2019-02-10 23:26:04.000000000 +0200
+++ libevent-2.1.12-stable/test/test.sh	2024-06-01 23:13:07.757375708 +0300
@@ -102,7 +102,7 @@
 	announce_n " regress: "
 	if test "$TEST_OUTPUT_FILE" = "/dev/null" ;
 	then
-		$TEST_DIR/regress --quiet $REGRESS_ARGS
+		$TEST_DIR/regress $REGRESS_ARGS
 	else
 		$TEST_DIR/regress $REGRESS_ARGS >>"$TEST_OUTPUT_FILE"
 	fi
@@ -117,7 +117,7 @@
 	announce_n " regress_debug: "
 	if test "$TEST_OUTPUT_FILE" = "/dev/null" ;
 	then
-		EVENT_DEBUG_MODE=1 $TEST_DIR/regress --quiet $REGRESS_ARGS
+		EVENT_DEBUG_MODE=1 $TEST_DIR/regress $REGRESS_ARGS
 	else
 		EVENT_DEBUG_MODE=1 $TEST_DIR/regress $REGRESS_ARGS >>"$TEST_OUTPUT_FILE"
 	fi
