--- libffi-3.5.2/configure.host	2025-08-02 10:29:39.000000000 +0400
+++ libffi-3.5.2/configure.host	2026-02-08 23:58:38.252060771 +0400
@@ -206,10 +206,9 @@
   powerpc-*-darwin* | powerpc64-*-darwin*)
 	TARGET=POWERPC_DARWIN; TARGETDIR=powerpc
 	;;
-  powerpc-*-aix* | rs6000-*-aix*)
+  powerpc*-*-aix* | rs6000-*-aix*)
 	TARGET=POWERPC_AIX; TARGETDIR=powerpc
-	# Create AIX-style "FAT" libraries.
-	tmake_file="t-aix"
+	# HAVE_LONG_DOUBLE_VARIANT=1
 	;;
   powerpc-*-freebsd* | powerpc-*-openbsd* | powerpc-*-netbsd*)
 	TARGET=POWERPC_FREEBSD; TARGETDIR=powerpc
