--- libidn2-2.3.8/tests/Makefile.in	2026-02-11 13:52:24.417318629 +0200
+++ libidn2-2.3.8/tests/Makefile.in	2026-02-11 13:52:43.907454318 +0200
@@ -211,7 +211,7 @@
 am__EXEEXT_2 = test-version$(EXEEXT) test-strerror$(EXEEXT) \
 	test-locale$(EXEEXT) test-tounicode$(EXEEXT) \
 	test-punycode$(EXEEXT) test-compat-punycode$(EXEEXT) \
-	test-IdnaTest-inc$(EXEEXT) test-IdnaTest-txt$(EXEEXT) \
+	test-IdnaTest-inc$(EXEEXT) \
 	test-lookup$(EXEEXT) test-register$(EXEEXT) $(am__EXEEXT_1)
 test_IdnaTest_inc_SOURCES = test-IdnaTest-inc.c
 test_IdnaTest_inc_OBJECTS = test-IdnaTest-inc.$(OBJEXT)
