--- libsodium-stable/test/default/Makefile.in	2024-04-04 13:53:05.015995158 -0500
+++ libsodium-stable/test/default/Makefile.in	2024-04-04 13:54:00.357186798 -0500
@@ -149,7 +149,7 @@
 	kdf$(EXEEXT) keygen$(EXEEXT) kx$(EXEEXT) metamorphic$(EXEEXT) \
 	misuse$(EXEEXT) onetimeauth$(EXEEXT) onetimeauth2$(EXEEXT) \
 	onetimeauth7$(EXEEXT) pwhash_argon2i$(EXEEXT) \
-	pwhash_argon2id$(EXEEXT) randombytes$(EXEEXT) \
+	pwhash_argon2id$(EXEEXT) \
 	scalarmult$(EXEEXT) scalarmult2$(EXEEXT) scalarmult5$(EXEEXT) \
 	scalarmult6$(EXEEXT) scalarmult7$(EXEEXT) scalarmult8$(EXEEXT) \
 	secretbox$(EXEEXT) secretbox2$(EXEEXT) secretbox7$(EXEEXT) \
