--- libunistring-1.4.1/tests/test-stat-time.c	2025-01-23 01:41:12.000000000 +0400
+++ libunistring-1.4.1/tests/test-stat-time.c	2026-02-11 18:44:41.951248474 +0400
@@ -141,22 +141,22 @@
   ASSERT (statinfo[0].st_mtime < statinfo[2].st_mtime
           || (statinfo[0].st_mtime == statinfo[2].st_mtime
               && (get_stat_mtime_ns (&statinfo[0])
-                  < get_stat_mtime_ns (&statinfo[2]))));
+                  <= get_stat_mtime_ns (&statinfo[2]))));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (statinfo[2].st_mtime < statinfo[3].st_mtime
           || (statinfo[2].st_mtime == statinfo[3].st_mtime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_mtime_ns (&statinfo[3]))));
+                  <= get_stat_mtime_ns (&statinfo[3]))));
 
   /* Now check the result of the access functions. */
   /* mtime(stamp1) < mtime(stamp2) */
   ASSERT (modtimes[0].tv_sec < modtimes[2].tv_sec
           || (modtimes[0].tv_sec == modtimes[2].tv_sec
-              && modtimes[0].tv_nsec < modtimes[2].tv_nsec));
+              && modtimes[0].tv_nsec <= modtimes[2].tv_nsec));
   /* mtime(stamp2) < mtime(stamp3) */
   ASSERT (modtimes[2].tv_sec < modtimes[3].tv_sec
           || (modtimes[2].tv_sec == modtimes[3].tv_sec
-              && modtimes[2].tv_nsec < modtimes[3].tv_nsec));
+              && modtimes[2].tv_nsec <= modtimes[3].tv_nsec));
 
   /* verify equivalence */
   for (i = 0; i < NFILES; ++i)
@@ -186,7 +186,7 @@
   ASSERT (statinfo[2].st_mtime < statinfo[1].st_ctime
           || (statinfo[2].st_mtime == statinfo[1].st_ctime
               && (get_stat_mtime_ns (&statinfo[2])
-                  < get_stat_ctime_ns (&statinfo[1]))));
+                  <= get_stat_ctime_ns (&statinfo[1]))));
 }
 #endif
 
