--- mpdecimal-4.0.1/configure	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/configure	2026-02-09 11:11:24.841990081 +0400
@@ -2832,13 +2832,10 @@
     CONFIGURE_LDFLAGS="-dynamiclib $FPIC -install_name \$(libdir)/$LIBSONAME -compatibility_version 4.0 -current_version 4.0.1"
     ;;
   *-*-aix*)
-    LIBNAME=
-    LIBSONAME=
-    LIBSHARED="shr.o"
-    LIBSHARED_USE_AR="yes"
-    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-bnoentry -Wl,-bE:.objs/symbols.exp"
-    LINK_STATIC="-Wl,-bstatic"
-    LINK_DYNAMIC="-Wl,-bshared"
+    LIBNAME="libmpdec.so"
+    LIBSONAME="libmpdec.so.4"
+    LIBSHARED="libmpdec.so.4.0.0"
+    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-G"
     ;;
   *-*-mingw*)
     FPIC=
@@ -3000,16 +2997,6 @@
   as_fn_error $? "at least one of --enable-static or --enable-shared must be set" "$LINENO" 5
 fi
 
-# AIX: both static and shared libraries must be enabled:
-case $host in
-  *-*-aix*)
-    if test x"$ENABLE_STATIC" = x"no" -o x"$ENABLE_SHARED" = x"no"
-    then
-      as_fn_error $? "the AIX build requires --enable-static and --enable-shared" "$LINENO" 5
-    fi
-    ;;
-esac
-
 # Check whether to install the documentation:
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for --enable-doc" >&5
 $as_echo_n "checking for --enable-doc... " >&6; }
--- mpdecimal-4.0.1/tests/Makefile.in	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/tests/Makefile.in	2026-02-09 11:17:41.053570208 +0400
@@ -40,12 +40,12 @@
 runtest:\
 Makefile runtest.c test.c $(SRCDIR)/$(LIBSTATIC) $(SRCDIR)/mpdecimal.h \
 test.h vctest.h
-	$(CC) -I$(SRCDIR) $(MPD_CFLAGS) -o runtest runtest.c test.c $(LINK_LIBSTATIC) -lm
+	$(CC) -I$(SRCDIR) $(MPD_CFLAGS) -o runtest runtest.c test.c $(LDFLAGS) -lm
 
 runtest_shared:\
 Makefile runtest.c test.c $(SRCDIR)/$(LIBSHARED) $(SRCDIR)/mpdecimal.h \
 test.h vctest.h
-	$(CC) -I$(SRCDIR) $(MPD_CFLAGS_SHARED) -o runtest_shared runtest.c test.c -L$(SRCDIR) -lmpdec -lm
+	$(CC) -I$(SRCDIR) $(MPD_CFLAGS_SHARED) -o runtest_shared runtest.c test.c $(LDFLAGS) -L$(SRCDIR) -lmpdec -lm
 
 
 FORCE:
--- mpdecimal-4.0.1/tests/gettests.sh	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/tests/gettests.sh	2026-02-19 11:18:00.103733912 +0400
@@ -1,23 +1,19 @@
 #!/bin/sh
 
+set -e
+
 if [ ! -d testdata ]; then
     mkdir testdata
 fi
 
 if [ x"$1" != x"--local" -a ! -f testdata/add.decTest ]; then
     # The official tests are freely downloadable, but copyrighted.
-    WGET=`which wget`
-    if [ ! -f "$WGET" ]; then
-        printf "\n*** gettests.sh: error: could not find wget. In order to run the tests, \n"
-        printf "    download http://speleotrove.com/decimal/dectest.zip and extract the \n"
-        printf "    archive into tests/testdata.\n\n"
-        exit 1
-    fi
     printf "\nGetting official tests ... \n\n"
-    $WGET -nv http://speleotrove.com/decimal/dectest.zip 2>&1 &&
-    cd testdata &&
-    unzip -qq ../dectest.zip &&
-    cd ../
+    curl --insecure --location --remote-name --silent --show-error --fail \
+        ftp://python.infinidat.com/python/sources/dectest.zip
+    cd testdata
+    python3 -m zipfile -e ../dectest.zip .
+    cd ..
 fi
 
 if [ ! -f testdata/baseconv.decTest ]; then
--- mpdecimal-4.0.1/tests/runshort.sh	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/tests/runshort.sh	2026-02-18 22:47:26.160132220 +0400
@@ -38,19 +38,22 @@
     printf "# ========================================================================\n\n"
 
     PORTABLE_PWD=`pwd`
-    LD_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec:$LD_LIBRARY_PATH"
-    DYLD_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec:$DYLD_LIBRARY_PATH"
-    LD_64_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec:$LD_64_LIBRARY_PATH"
-    LD_32_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec:$LD_32_LIBRARY_PATH"
-    LD_LIBRARY_PATH_64="$PORTABLE_PWD/../libmpdec:$LD_LIBRARY_PATH_64"
-    LD_LIBRARY_PATH_32="$PORTABLE_PWD/../libmpdec:$LD_LIBRARY_PATH_32"
-    PATH="$LD_LIBRARY_PATH:$PATH"
-    export LD_LIBRARY_PATH
-    export DYLD_LIBRARY_PATH
-    export LD_64_LIBRARY_PATH
-    export LD_32_LIBRARY_PATH
-    export LD_LIBRARY_PATH_64
-    export LD_LIBRARY_PATH_32
+
+    if [ -n "$LD_LIBRARY_PATH" ]; then
+        LD_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec:$LD_LIBRARY_PATH"
+    else
+        LD_LIBRARY_PATH="$PORTABLE_PWD/../libmpdec"
+    fi
+
+    if [ -n "$LIBPATH" ]; then
+        LIBPATH="$PORTABLE_PWD/../libmpdec:$LIBPATH"
+    else
+        LIBPATH="$PORTABLE_PWD/../libmpdec"
+    fi
+
+    PATH="$PORTABLE_PWD/../libmpdec:$PATH"
+
+    export LD_LIBRARY_PATH LIBPATH PATH
 
     if [ x"$1" != x"--local" ]; then
         printf "Running official tests ...\n\n"
