--- mpdecimal-4.0.1/configure	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/configure	2026-02-09 11:11:24.841990081 +0400
@@ -2832,13 +2832,10 @@
     CONFIGURE_LDFLAGS="-dynamiclib $FPIC -install_name \$(libdir)/$LIBSONAME -compatibility_version 4.0 -current_version 4.0.1"
     ;;
   *-*-aix*)
-    LIBNAME=
-    LIBSONAME=
-    LIBSHARED="shr.o"
-    LIBSHARED_USE_AR="yes"
-    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-bnoentry -Wl,-bE:.objs/symbols.exp"
-    LINK_STATIC="-Wl,-bstatic"
-    LINK_DYNAMIC="-Wl,-bshared"
+    LIBNAME="libmpdec.so"
+    LIBSONAME="libmpdec.so.4"
+    LIBSHARED="libmpdec.so.4.0.0"
+    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-G"
     ;;
   *-*-mingw*)
     FPIC=
@@ -3000,16 +2997,6 @@
   as_fn_error $? "at least one of --enable-static or --enable-shared must be set" "$LINENO" 5
 fi
 
-# AIX: both static and shared libraries must be enabled:
-case $host in
-  *-*-aix*)
-    if test x"$ENABLE_STATIC" = x"no" -o x"$ENABLE_SHARED" = x"no"
-    then
-      as_fn_error $? "the AIX build requires --enable-static and --enable-shared" "$LINENO" 5
-    fi
-    ;;
-esac
-
 # Check whether to install the documentation:
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for --enable-doc" >&5
 $as_echo_n "checking for --enable-doc... " >&6; }
Only in mpdecimal-4.0.1: configure.orig
--- mpdecimal-4.0.1/tests/Makefile.in	2025-05-05 02:25:00.000000000 +0400
+++ mpdecimal-4.0.1/tests/Makefile.in	2026-02-09 11:17:41.053570208 +0400
@@ -40,12 +40,12 @@
 runtest:\
 Makefile runtest.c test.c $(SRCDIR)/$(LIBSTATIC) $(SRCDIR)/mpdecimal.h \
 test.h vctest.h
-	$(CC) -I$(SRCDIR) $(MPD_CFLAGS) -o runtest runtest.c test.c $(LINK_LIBSTATIC) -lm
+	$(CC) -I$(SRCDIR) $(MPD_CFLAGS) -o runtest runtest.c test.c $(LDFLAGS) -lm
 
 runtest_shared:\
 Makefile runtest.c test.c $(SRCDIR)/$(LIBSHARED) $(SRCDIR)/mpdecimal.h \
 test.h vctest.h
-	$(CC) -I$(SRCDIR) $(MPD_CFLAGS_SHARED) -o runtest_shared runtest.c test.c -L$(SRCDIR) -lmpdec -lm
+	$(CC) -I$(SRCDIR) $(MPD_CFLAGS_SHARED) -o runtest_shared runtest.c test.c $(LDFLAGS) -L$(SRCDIR) -lmpdec -lm
 
 
 FORCE:
