--- mpdecimal-4.0.0/configure	2024-01-09 18:25:00.000000000 -0600
+++ mpdecimal-4.0.0/configure	2024-04-04 22:42:00.239004425 -0500
@@ -2846,13 +2846,10 @@
     CONFIGURE_LDFLAGS="-dynamiclib $FPIC -install_name @rpath/$LIBSONAME -compatibility_version 4.0 -current_version 4.0.0"
     ;;
   *-*-aix*)
-    LIBNAME=
-    LIBSONAME=
-    LIBSHARED="shr.o"
-    LIBSHARED_USE_AR="yes"
-    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-bnoentry -Wl,-bE:.objs/symbols.exp"
-    LINK_STATIC="-Wl,-bstatic"
-    LINK_DYNAMIC="-Wl,-bshared"
+    LIBNAME="libmpdec.so"
+    LIBSONAME="libmpdec.so.4"
+    LIBSHARED="libmpdec.so.4.0.0"
+    CONFIGURE_LDFLAGS="-shared $FPIC -Wl,-G"
     ;;
   *-*-mingw*)
     FPIC=
@@ -2998,16 +2995,6 @@
   as_fn_error $? "at least one of --enable-static or --enable-shared must be set" "$LINENO" 5
 fi
 
-# AIX: both static and shared libraries must be enabled:
-case $host in
-  *-*-aix*)
-    if test x"$ENABLE_STATIC" = x"no" -o x"$ENABLE_SHARED" = x"no"
-    then
-      as_fn_error $? "the AIX build requires --enable-static and --enable-shared" "$LINENO" 5
-    fi
-    ;;
-esac
-
 # Check whether to install the documentation:
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for --enable-doc" >&5
 $as_echo_n "checking for --enable-doc... " >&6; }
