--- nettle-3.10.2/Makefile.in	2025-06-26 22:29:02.000000000 +0400
+++ nettle-3.10.2/Makefile.in	2026-02-03 13:06:50.580406161 +0400
@@ -497,7 +497,7 @@
 
 install-static: $(LIBTARGETS)
 	$(MKDIR_P) $(DESTDIR)$(libdir)
-	for f in $(LIBTARGETS); do \
+	test -z "$(LIBTARGETS)" || for f in "$(LIBTARGETS)"; do \
 	  $(INSTALL_DATA) $$f $(DESTDIR)$(libdir) ; \
 	done
 
--- nettle-3.10.2/configure	2025-06-26 22:29:02.000000000 +0400
+++ nettle-3.10.2/configure	2026-02-03 13:07:07.261890564 +0400
@@ -8067,6 +8067,19 @@
     LIBHOGWEED_LINK='$(CC) $(CFLAGS) $(LDFLAGS) --shared -h $(LIBHOGWEED_SONAME)'
     LIBHOGWEED_LIBS='libnettle.so $(LIBS)'
     ;;
+  aix*)
+    LIBNETTLE_FORLINK=libnettle.so
+    LIBNETTLE_SONAME='$(LIBNETTLE_FORLINK).$(LIBNETTLE_MAJOR)'
+    LIBNETTLE_FILE='$(LIBNETTLE_SONAME).$(LIBNETTLE_MINOR)'
+    LIBNETTLE_LINK='$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,-G'
+    LIBNETTLE_LIBS=''
+
+    LIBHOGWEED_FORLINK=libhogweed.so
+    LIBHOGWEED_SONAME='$(LIBHOGWEED_FORLINK).$(LIBHOGWEED_MAJOR)'
+    LIBHOGWEED_FILE='$(LIBHOGWEED_SONAME).$(LIBHOGWEED_MINOR)'
+    LIBHOGWEED_LINK='$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,-G'
+    LIBHOGWEED_LIBS='libnettle.so $(LIBS)'
+    ;;
   *)
     LIBNETTLE_FORLINK=libnettle.so
     LIBNETTLE_SONAME='$(LIBNETTLE_FORLINK).$(LIBNETTLE_MAJOR)'
@@ -8788,7 +8801,7 @@
 
 # Set these flags *last*, or else the test programs won't compile
 if test x$GCC = xyes ; then
-  CFLAGS="$CFLAGS -ggdb3 -Wall -W -Wno-sign-compare \
+  CFLAGS="$CFLAGS -Wall -W -Wno-sign-compare \
   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes \
   -Wpointer-arith -Wbad-function-cast -Wnested-externs"
 
