--- openssl-1.1.1w/Configurations/10-main.conf	2023-10-15 17:03:49.983897240 -0500
+++ openssl-1.1.1w/Configurations/10-main.conf	2023-10-15 17:04:48.370698600 -0500
@@ -1136,7 +1136,7 @@
         template         => 1,
         sys_id           => "AIX",
         lib_cppflags     => "-DB_ENDIAN",
-        lflags           => "-Wl,-bsvr4",
+        lflags           => "-Wl,-bsvr4 -Wl,-b64",
         thread_scheme    => "pthreads",
         dso_scheme       => "dlfcn",
         shared_target    => "self",
@@ -1144,9 +1144,6 @@
         shared_ldflag    => "-Wl,-G,-bsymbolic,-bnoentry",
         shared_defflag   => "-Wl,-bE:",
         shared_extension => ".so.\$(SHLIB_VERSION_NUMBER)",
-        dso_extension    => ".so",
-        lib_extension    => shared("_a.a"),
-        shared_extension_simple => shared(".a"),
     },
     "aix-gcc" => {
         inherit_from     => [ "aix-common", asm("ppc32_asm") ],
@@ -1171,7 +1168,6 @@
         bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
         perlasm_scheme   => "aix64",
         shared_ldflag    => add_before("-shared -static-libgcc"),
-        shared_extension => "64.so.\$(SHLIB_VERSION_NUMBER)",
         AR               => add("-X64"),
         RANLIB           => add("-X64"),
     },
