--- openssl-1.1.1zh/Configurations/10-main.conf	2026-06-16 09:33:01.000000000 +0400
+++ openssl-1.1.1zh/Configurations/10-main.conf	2026-08-15 16:14:45.308281637 +0400
@@ -226,13 +226,13 @@
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3 -fomit-frame-pointer")),
+                                              release => "-O2 -fomit-frame-pointer")),
         cflags           => add(threads("-pthread")),
         lib_cppflags     => add("-DL_ENDIAN"),
         ex_libs          => add(threads("-pthread")),
         bn_ops           => "BN_LLONG",
         shared_cflag     => "-fPIC",
-        shared_ldflag    => add_before("-shared -static-libgcc"),
+        shared_ldflag    => add_before("-shared"),
     },
     "solaris64-x86_64-gcc" => {
         # -shared -static-libgcc might appear controversial, but modules
@@ -247,15 +247,14 @@
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3")),
+                                              release => "-O2")),
         cflags           => add_before("-m64", threads("-pthread")),
         lib_cppflags     => add("-DL_ENDIAN"),
         ex_libs          => add(threads("-pthread")),
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         perlasm_scheme   => "elf",
         shared_cflag     => "-fPIC",
-        shared_ldflag    => add_before("-shared -static-libgcc"),
-        multilib         => "/64",
+        shared_ldflag    => add_before("-shared"),
     },
 
 #### Solaris x86 with Sun C setups
@@ -295,7 +294,7 @@
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3")),
+                                              release => "-O2")),
         cflags           => add(threads("-pthread")),
         lib_cppflags     => add("-DB_ENDIAN -DBN_DIV2W"),
         ex_libs          => add(threads("-pthread")),
@@ -311,13 +310,12 @@
         # -m32 should be safe to add as long as driver recognizes
         # -mcpu=ultrasparc
         inherit_from     => [ "solaris-sparcv7-gcc", asm("sparcv9_asm") ],
-        cflags           => add_before("-m32 -mcpu=ultrasparc"),
+        cflags           => add_before("-m32"),
     },
     "solaris64-sparcv9-gcc" => {
         inherit_from     => [ "solaris-sparcv9-gcc" ],
         cflags           => sub { my $f=join(" ",@_); $f =~ s/\-m32/-m64/; $f; },
         bn_ops           => "BN_LLONG RC4_CHAR",
-        multilib         => "/64",
     },
 
 #### SPARC Solaris with Sun C setups
@@ -1136,7 +1134,7 @@
         template         => 1,
         sys_id           => "AIX",
         lib_cppflags     => "-DB_ENDIAN",
-        lflags           => "-Wl,-bsvr4",
+        lflags           => "-Wl,-bsvr4 -Wl,-b64 -Wl,-brtl",
         thread_scheme    => "pthreads",
         dso_scheme       => "dlfcn",
         shared_target    => "self",
@@ -1144,9 +1142,6 @@
         shared_ldflag    => "-Wl,-G,-bsymbolic,-bnoentry",
         shared_defflag   => "-Wl,-bE:",
         shared_extension => ".so.\$(SHLIB_VERSION_NUMBER)",
-        dso_extension    => ".so",
-        lib_extension    => shared("_a.a"),
-        shared_extension_simple => shared(".a"),
     },
     "aix-gcc" => {
         inherit_from     => [ "aix-common", asm("ppc32_asm") ],
@@ -1170,8 +1165,7 @@
         ex_libs          => add(threads("-pthread")),
         bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
         perlasm_scheme   => "aix64",
-        shared_ldflag    => add_before("-shared -static-libgcc"),
-        shared_extension => "64.so.\$(SHLIB_VERSION_NUMBER)",
+        shared_ldflag    => add_before("-shared"),
         AR               => add("-X64"),
         RANLIB           => add("-X64"),
     },
--- openssl-1.1.1zh/Configurations/unix-Makefile.tmpl	2026-06-16 09:33:01.000000000 +0400
+++ openssl-1.1.1zh/Configurations/unix-Makefile.tmpl	2026-08-15 16:14:45.310157191 +0400
@@ -38,7 +38,7 @@
 
      # Shared AIX support is special. We put libcrypto[64].so.ver into
      # libcrypto.a and use libcrypto_a.a as static one.
-     sub sharedaix  { !$disabled{shared} && $config{target} =~ /^aix/ }
+     sub sharedaix  { !$disabled{shared} && $config{target} =~ /^aix-cc/ }
 
      our $sover_dirname = $config{shlib_version_number};
      $sover_dirname =~ s|\.|_|g
@@ -134,7 +134,6 @@
 GENERATED={- # common0.tmpl provides @generated
              join(" ", @generated ) -}
 
-INSTALL_LIBS={- join(" ", map { lib($_) } @{$unified_info{install}->{libraries}}) -}
 INSTALL_SHLIBS={- join(" ", map { shlib($_) } @{$unified_info{install}->{libraries}}) -}
 INSTALL_SHLIB_INFO={- join(" ", map { "\"".shlib($_).";".shlib_simple($_)."\"" } @{$unified_info{install}->{libraries}}) -}
 INSTALL_ENGINES={- join(" ", map { dso($_) } @{$unified_info{install}->{engines}}) -}
@@ -191,7 +190,7 @@
 # $(libdir) is chosen to be compatible with the GNU coding standards
 libdir={- file_name_is_absolute($libdir)
           ? $libdir : '$(INSTALLTOP)/$(LIBDIR)' -}
-ENGINESDIR=$(libdir)/engines-{- $sover_dirname -}
+ENGINESDIR=$(libdir)/engines
 
 # Convenience variable for those who want to set the rpath in shared
 # libraries and applications
@@ -245,7 +244,7 @@
 ENGINESDIR_dev={- use File::Spec::Win32;
                   our $enginesdir =
                       File::Spec::Win32->catdir($prefix,$libdir,
-                                                "engines-$sover_dirname");
+                                                "engines");
                   our ($enginesdir_dev, $enginesdir_dir, $enginesdir_file) =
                       File::Spec::Win32->splitpath($enginesdir, 1);
                   $enginesdir =~ s|\\|/|g;
@@ -610,15 +609,6 @@
 		chmod 644 "$(DESTDIR)$(INSTALLTOP)/include/openssl/$$fn"; \
 	done
 	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(libdir)"
-	@set -e; for l in $(INSTALL_LIBS); do \
-		fn=`basename $$l`; \
-		$(ECHO) "install $$l -> $(DESTDIR)$(libdir)/$$fn"; \
-		cp $$l "$(DESTDIR)$(libdir)/$$fn.new"; \
-		$(RANLIB) "$(DESTDIR)$(libdir)/$$fn.new"; \
-		chmod 644 "$(DESTDIR)$(libdir)/$$fn.new"; \
-		mv -f "$(DESTDIR)$(libdir)/$$fn.new" \
-		      "$(DESTDIR)$(libdir)/$$fn"; \
-	done
 	@ : {- output_off() if $disabled{shared}; "" -}
 	@set -e; for s in $(INSTALL_SHLIB_INFO); do \
 		s1=`echo "$$s" | cut -f1 -d";"`; \
@@ -959,7 +949,7 @@
 	        echo 'libdir=$(libdir)'; \
 	    fi; \
 	    echo 'includedir=$${prefix}/include'; \
-	    echo 'enginesdir=$${libdir}/engines-{- $sover_dirname -}'; \
+	    echo 'enginesdir=$${libdir}/engines'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL-libcrypto'; \
 	    echo 'Description: OpenSSL cryptography library'; \
--- openssl-1.1.1zh/Configurations/windows-makefile.tmpl	2026-06-16 09:33:01.000000000 +0400
+++ openssl-1.1.1zh/Configurations/windows-makefile.tmpl	2026-08-15 16:14:45.310879945 +0400
@@ -144,7 +144,7 @@
 LIBDIR={- our $libdir = $config{libdir} || "lib";
           file_name_is_absolute($libdir) ? "" : $libdir -}
 ENGINESDIR_dev={- use File::Spec::Functions qw(:DEFAULT splitpath);
-                  our $enginesdir = catdir($prefix,$libdir,"engines-$sover_dirname");
+                  our $enginesdir = catdir($prefix,$libdir,"engines");
                   our ($enginesdir_dev, $enginesdir_dir, $enginesdir_file) =
                       splitpath($enginesdir, 1);
                   $enginesdir_dev -}
@@ -251,7 +251,7 @@
                 join(' ', $lib_cppflags,
                           (map { '-D'.quotify1($_) }
                                "OPENSSLDIR=\"$openssldir\"",
-                               "ENGINESDIR=\"$enginesdir\""),
+                               "ENGINESDIR=\"$openssldir\""),
                           '$(CNF_CPPFLAGS)', '$(CPPFLAGS)') -}
 LIB_CFLAGS={- join(' ', $target{lib_cflags} || (),
                         $target{shared_cflag} || (),
--- openssl-1.1.1zh/build.info	2026-06-16 09:33:01.000000000 +0400
+++ openssl-1.1.1zh/build.info	2026-08-15 16:14:45.311212072 +0400
@@ -86,8 +86,8 @@
  SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
  SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
 ELSIF[{- $config{target} =~ /^VC-/ -}]
- SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $target{multilib} -}
- SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $target{multilib} -}
+ SHARED_NAME[libcrypto]=crypto
+ SHARED_NAME[libssl]=ssl
 ENDIF
 
 # VMS has a cultural standard where all libraries are prefixed.
--- openssl-1.1.1zh/ms/uplink.c	2026-06-16 09:33:01.000000000 +0400
+++ openssl-1.1.1zh/ms/uplink.c	2026-08-15 16:29:22.382178811 +0400
@@ -60,10 +60,14 @@
         _tcscpy(msg + len, _T("unimplemented function"));
 
         if ((h = apphandle) == NULL) {
-            if ((h = GetModuleHandle(NULL)) == NULL) {
-                apphandle = (HMODULE) - 1;
-                _tcscpy(msg + len, _T("no host application"));
-                break;
+            if ((h = GetModuleHandleA("_ssl.pyd")) == NULL) {
+                if ((h = GetModuleHandleA("_ssl_d.pyd")) == NULL) {
+                    if ((h = GetModuleHandle(NULL)) == NULL) {
+                        apphandle = (HMODULE) - 1;
+                        _tcscpy(msg + len, _T("no host application"));
+                        break;
+                    }
+                }
             }
             apphandle = h;
         }
