diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index ba48173..64c2d5e 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -432,7 +432,19 @@ def _init_posix():

     global _config_vars
     _config_vars = g
-
+    _fix_prefix(_config_vars)
+
+def _fix_prefix(config_vars):
+    import sys
+    if config_vars is None:
+        return
+    makefile_prefix = config_vars.get('prefix', None)
+    if not makefile_prefix:
+        return
+    for key, value in config_vars.items():
+        if not isinstance(value, str):
+            continue
+        config_vars[key] = value.replace(makefile_prefix, sys.prefix)

 def _init_nt():
     """Initialize the module as appropriate for NT"""
