--- Python-3.11.14/Lib/test/test_importlib/test_threaded_import.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_importlib/test_threaded_import.py	2026-03-06 22:25:12.526912512 +0400
@@ -254,6 +254,7 @@
                           'partial', 'cfimport.py')
         script_helper.assert_python_ok(fn)
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue43665')
     def test_multiprocessing_pool_circular_import(self):
         # Regression test for bpo-41567
         fn = os.path.join(os.path.dirname(__file__),
