diff --git a/PC/getpathp.c b/PC/getpathp.c
index 04f24d9..68a4970 100644
--- a/PC/getpathp.c
+++ b/PC/getpathp.c
@@ -821,10 +821,6 @@ calculate_module_search_path(PyCalculatePath *calculate,
                              const wchar_t *zip_path)
 {
     int skiphome = calculate->home==NULL ? 0 : 1;
-#ifdef Py_ENABLE_SHARED
-    calculate->machine_path = getpythonregpath(HKEY_LOCAL_MACHINE, skiphome);
-    calculate->user_path = getpythonregpath(HKEY_CURRENT_USER, skiphome);
-#endif
     /* We only use the default relative PYTHONPATH if we haven't
        anything better to use! */
     int skipdefault = (calculate->pythonpath_env != NULL ||
