--- Python-3.11.14/Lib/test/_test_eintr.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/_test_eintr.py	2026-03-06 22:23:42.220026307 +0400
@@ -531,6 +531,9 @@
     def test_lockf(self):
         self._lock(fcntl.lockf, "lockf")
 
+    # Issue 35633: See https://bugs.python.org/issue35633#msg333662
+    # skip test rather than accept PermissionError from all platforms
+    @unittest.skipIf(platform.system() == "AIX", "AIX returns PermissionError")
     def test_flock(self):
         self._lock(fcntl.flock, "flock")
 
