--- Python-3.11.14/Lib/test/test_smtpnet.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_smtpnet.py	2026-03-06 22:24:20.381235820 +0400
@@ -25,6 +25,7 @@
     testServer = 'smtp.gmail.com'
     remotePort = 587
 
+    @unittest.skip('HOSTDEV-3483')
     def test_connect_starttls(self):
         support.get_attribute(smtplib, 'SMTP_SSL')
         context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
