--- Python-3.8.0/setup.py       2019-10-14 16:34:47.000000000 +0300
+++ Python-3.8.0/setup.py       2019-12-04 12:56:15.752076764 +0200
@@ -1282,7 +1285,7 @@
                             ndbm_libs = ['ndbm']
                         elif self.compiler.find_library_file(self.lib_dirs,
                                                              'gdbm_compat'):
-                            ndbm_libs = ['gdbm_compat']
+                            ndbm_libs = ['gdbm_compat', 'gdbm']
                         else:
                             ndbm_libs = []
                         if dbm_setup_debug: print("building dbm using ndbm")

