--- Python-3.8.0/setup.py       2019-10-14 16:34:47.000000000 +0300
+++ Python-3.8.0/setup_changed.py       2019-12-04 12:32:01.575237649 +0200
@@ -1032,7 +1032,10 @@
             # the encryption.
             return
 
-        if self.compiler.find_library_file(self.lib_dirs, 'crypt'):
+
+        if self.compiler.find_library_file(self.lib_dirs, 'crypt_i'):
+            libs = ['crypt_i']
+        elif self.compiler.find_library_file(self.lib_dirs, 'crypt'):
             libs = ['crypt']
         else:
             libs = []
