Index: cpython/Lib/sysconfig.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/Lib/sysconfig.py	(date 1529244314000)
+++ cpython/Lib/sysconfig.py	(date 1529244314000)
@@ -416,6 +416,7 @@
     from distutils.sysconfig import _fix_prefix
     build_time_vars = _temp.build_time_vars
     vars.update(build_time_vars)
+    _fix_prefix(vars)
 
 def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
