--- Python-3.11.16/Lib/test/test_platform.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_platform.py	2026-03-11 00:13:45.825034097 +0400
@@ -337,10 +337,7 @@
         if csd:
             self.assertTrue(csd.startswith('SP'), msg=csd)
         if ptype:
-            if os.cpu_count() > 1:
-                self.assertIn('Multiprocessor', ptype)
-            else:
-                self.assertIn('Uniprocessor', ptype)
+            self.assertIn(ptype, ('Multiprocessor Free', 'Uniprocessor Free'))
 
     @unittest.skipIf(support.MS_WINDOWS, 'This test only makes sense on non Windows')
     def test_win32_ver_on_non_windows(self):
