--- Python-3.11.16/Lib/test/test_openpty.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_openpty.py	2026-08-11 17:41:16.793622924 +0400
@@ -1,6 +1,10 @@
 # Test to see if openpty works. (But don't worry if it isn't available.)
 
-import os, unittest
+import os
+import platform
+import signal
+import sys
+import unittest
 
 if not hasattr(os, "openpty"):
     raise unittest.SkipTest("os.openpty() not available.")
@@ -8,6 +12,9 @@
 
 class OpenptyTest(unittest.TestCase):
     def test(self):
+        if (platform.system() == "SunOS" and platform.release() == "5.10"):
+            old_sighup = signal.signal(signal.SIGHUP, signal.SIG_IGN)
+            self.addCleanup(signal.signal, signal.SIGHUP, old_sighup)
         master, slave = os.openpty()
         self.addCleanup(os.close, master)
         self.addCleanup(os.close, slave)
