Index: cpython/PC/getpathp.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/PC/getpathp.c	(date 1529244875000)
+++ cpython/PC/getpathp.c	(date 1529244875000)
@@ -759,10 +759,7 @@
                              wchar_t *prefix)
 {
     int skiphome = calculate->home==NULL ? 0 : 1;
-#ifdef Py_ENABLE_SHARED
-    calculate->machine_path = getpythonregpath(HKEY_LOCAL_MACHINE, skiphome);
-    calculate->user_path = getpythonregpath(HKEY_CURRENT_USER, skiphome);
-#endif
+
     /* We only use the default relative PYTHONPATH if we haven't
        anything better to use! */
     int skipdefault = (core_config->module_search_path_env != NULL ||
