--- Python-3.11.15/Lib/test/test_termios.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/test/test_termios.py	2026-08-11 17:54:52.049509075 +0400
@@ -1,5 +1,9 @@
 import errno
+import fcntl
 import os
+import platform
+import signal
+import struct
 import sys
 import tempfile
 import unittest
@@ -12,9 +16,18 @@
 class TestFunctions(unittest.TestCase):
 
     def setUp(self):
+        if (platform.system() == "SunOS" and platform.release() == "5.10"):
+            old_sighup = signal.signal(signal.SIGHUP, signal.SIG_IGN)
+            self.addCleanup(signal.signal, signal.SIGHUP, old_sighup)
         master_fd, self.fd = os.openpty()
         self.addCleanup(os.close, master_fd)
         self.stream = self.enterContext(open(self.fd, 'wb', buffering=0))
+        if platform.system() == "SunOS":
+            fcntl.ioctl(
+                self.fd,
+                termios.TIOCSWINSZ,
+                struct.pack('HHHH', 24, 80, 0, 0),
+            )
         tmp = self.enterContext(tempfile.TemporaryFile(mode='wb', buffering=0))
         self.bad_fd = tmp.fileno()
 
