--- Python-3.11.9/setup.py	2024-06-09 17:31:57.903793911 +0300
+++ Python-3.11.9/setup.py	2024-06-09 17:32:27.826715069 +0300
@@ -1476,6 +1476,7 @@
             Extension(
                 '_decimal',
                 ['_decimal/_decimal.c'],
+                libraries=['m'],
                 # Uncomment for extra functionality:
                 # define_macros=[('EXTRA_FUNCTIONALITY', 1)]
             )
--- Python-3.11.9/Lib/_aix_support.py	2024-04-02 12:25:04
+++ Python-3.11.9/Lib/_aix_support.py	2024-06-24 19:28:30
@@ -1,16 +1,10 @@
 """Shared AIX support functions."""
 
+import os
 import sys
 import sysconfig
 
-try:
-    import subprocess
-except ImportError:  # pragma: no cover
-    # _aix_support is used in distutils by setup.py to build C extensions,
-    # before subprocess dependencies like _posixsubprocess are available.
-    import _bootsubprocess as subprocess
 
-
 def _aix_tag(vrtl, bd):
     # type: (List[int], int) -> str
     # Infer the ABI bitwidth from maxsize (assuming 64 bit as the default)
@@ -35,12 +29,10 @@
     builddate reflect the current ABI levels of the runtime environment.
     If no builddate is found give a value that will satisfy pep425 related queries
     """
-    # All AIX systems to have lslpp installed in this location
-    out = subprocess.check_output(["/usr/bin/lslpp", "-Lqc", "bos.rte"])
-    out = out.decode("utf-8")
-    out = out.strip().split(":")  # type: ignore
-    _bd = int(out[-1]) if out[-1] != '' else 9988
-    return (str(out[2]), _bd)
+    uname = os.uname()
+    _rv = '%s.%s.%s.%s' % (uname.version, uname.release, 0, 0)
+    _bd = 9988
+    return (_rv, _bd)
 
 
 def aix_platform():
