--- Python-3.11.15/Lib/test/test_tarfile.py	2026-03-14 14:37:13.923913295 +0400
+++ Python-3.11.15/Lib/test/test_tarfile.py	2026-03-14 14:38:52.409593491 +0400
@@ -3614,6 +3614,7 @@
             with self.check_context(arc.open(), 'data'):
                 self.expect_file('parent/evil')
 
+    @unittest.skipIf(sys.platform == 'aix', 'Known issue: OSError [Errno 86] File name too long')
     @symlink_test
     @os_helper.skip_unless_symlink
     def test_realpath_limit_attack(self):
