diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 33b2791..0619cbe 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -317,6 +317,9 @@ def _init_posix(vars):
     # the scripts are in another directory.
     if _PYTHON_BUILD:
         vars['LDSHARED'] = vars['BLDSHARED']
+    global _CONFIG_VARS
+    from distutils.sysconfig import _fix_prefix
+    _fix_prefix(_CONFIG_VARS)
 
 def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
