--- Python-3.11.15/Lib/test/test_shutil.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/test/test_shutil.py	2026-08-10 18:38:11.211049676 +0400
@@ -2639,7 +2639,7 @@
                 self.assertRaises(OSError, self.zerocopy_fun, src, dst)
 
 
-@unittest.skipIf(not SUPPORTS_SENDFILE, 'os.sendfile() not supported')
+@unittest.skipUnless(shutil._USE_CP_SENDFILE, "shutil sendfile fast-copy is not supported")
 class TestZeroCopySendfile(_ZeroCopyFileTest, unittest.TestCase):
     PATCHPOINT = "os.sendfile"
 
