--- Python-3.11.15/Lib/encodings/aliases.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/encodings/aliases.py	2026-03-11 00:09:05.465516066 +0400
@@ -209,6 +209,11 @@
     'ms932'              : 'cp932',
     'mskanji'            : 'cp932',
     'ms_kanji'           : 'cp932',
+    '943'                : 'cp932',
+    'cp943'              : 'cp932',
+    'cp_943'             : 'cp932',
+    'ibm943'             : 'cp932',
+    'ibm_943'            : 'cp932',
 
     # cp949 codec
     '949'                : 'cp949',
