--- Python-3.8.12/Lib/encodings/aliases.py	2024-06-03 11:19:39.403298027 -0500
+++ Python-3.8.12/Lib/encodings/aliases.py	2024-06-03 11:22:13.158812791 -0500
@@ -209,6 +209,11 @@
     'ms932'              : 'cp932',
     'mskanji'            : 'cp932',
     'ms_kanji'           : 'cp932',
+    '943'                : 'cp932',
+    'cp943'              : 'cp932',
+    'cp_943'             : 'cp932',
+    'ibm943'             : 'cp932',
+    'ibm_943'            : 'cp932',
 
     # cp949 codec
     '949'                : 'cp949',
