--- Python-3.11.16/Lib/test/test_glob.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_glob.py	2026-08-11 15:58:06.380238697 +0400
@@ -46,7 +46,7 @@
         if self.dir_fd is not None:
             os.close(self.dir_fd)
         if {os.open, os.stat} <= os.supports_dir_fd and os.scandir in os.supports_fd:
-            self.dir_fd = os.open(self.tempdir, os.O_RDONLY | os.O_DIRECTORY)
+            self.dir_fd = os.open(self.tempdir, os.O_RDONLY | getattr(os, 'O_DIRECTORY', 0))
         else:
             self.dir_fd = None
 
