diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index 9e74ccd..909d0bb 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -187,7 +187,7 @@ elif os.name == "posix":
             with contextlib.closing(os.popen(cmd)) as f:
                 for line in f.readlines():
                     line = line.strip()
-                    if line.startswith('Default Library Path (ELF):'):
+                    if line.startswith('Default Library Path'):
                         paths = line.split()[4]

             if not paths: