Index: cpython/configure.ac
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/configure.ac	(date 1534346388000)
+++ cpython/configure.ac	(date 1534346388000)
@@ -2799,7 +2799,11 @@
 AC_CHECK_LIB(intl, textdomain,
 	[AC_DEFINE(WITH_LIBINTL, 1,
 	[Define to 1 if libintl is needed for locale functions.])
-        LIBS="-lintl $LIBS"])
+        LIBS="-lintl $LIBS"],
+        [AC_SEARCH_LIBS(textdomain, intl,
+                [AC_DEFINE(WITH_LIBINTL, 1,
+                [Define to 1 if libintl is needed for locale functions.])
+                LIBS="-lintl -liconv $LIBS"], , "-liconv")])
 
 # checks for system dependent C++ extensions support
 case "$ac_sys_system" in
