--- Python-3.11.16/Lib/test/test_concurrent_futures/util.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_concurrent_futures/util.py	2026-08-11 12:20:47.515612167 +0400
@@ -111,6 +111,8 @@
             self.skipTest("ProcessPoolExecutor unavailable on this system")
         if sys.platform == "win32":
             self.skipTest("require unix system")
+        if "forkserver" not in multiprocessing.get_all_start_methods():
+            self.skipTest("forkserver start method not available")
         return super().get_context()
 
 
