--- Python-3.11.16/Lib/test/test_os.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_os.py	2026-08-11 17:43:55.425836619 +0400
@@ -13,6 +13,7 @@
 import locale
 import os
 import pickle
+import platform
 import select
 import shutil
 import signal
@@ -4111,6 +4112,9 @@
 
     @unittest.skipUnless(hasattr(os, 'openpty'), "need os.openpty()")
     def test_openpty(self):
+        if (platform.system() == "SunOS" and platform.release() == "5.10"):
+            old_sighup = signal.signal(signal.SIGHUP, signal.SIG_IGN)
+            self.addCleanup(signal.signal, signal.SIGHUP, old_sighup)
         master_fd, slave_fd = os.openpty()
         self.addCleanup(os.close, master_fd)
         self.addCleanup(os.close, slave_fd)
