--- Python-3.11.15/Lib/test/test_getpath.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/test/test_getpath.py	2026-08-17 00:30:14.494983711 +0400
@@ -109,6 +109,7 @@
         actual = getpath(ns, expected)
         self.assertEqual(expected, actual)
 
+    @unittest.skip('HOSTDEV-3483')
     def test_registry_win32(self):
         """Test registry lookup on Windows.
 
--- Python-3.11.15/Modules/getpath.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Modules/getpath.py	2026-08-17 00:30:14.496736663 +0400
@@ -673,36 +673,6 @@
     else:
         pythonpath.append(joinpath(prefix, ZIP_LANDMARK))
 
-    if os_name == 'nt' and use_environment and winreg:
-        # QUIRK: Windows also lists paths in the registry. Paths are stored
-        # as the default value of each subkey of
-        # {HKCU,HKLM}\Software\Python\PythonCore\{winver}\PythonPath
-        # where winver is sys.winver (typically '3.x' or '3.x-32')
-        for hk in (winreg.HKEY_CURRENT_USER, winreg.HKEY_LOCAL_MACHINE):
-            try:
-                key = winreg.OpenKeyEx(hk, WINREG_KEY)
-                try:
-                    i = 0
-                    while True:
-                        try:
-                            v = winreg.QueryValue(key, winreg.EnumKey(key, i))
-                        except OSError:
-                            break
-                        if isinstance(v, str):
-                            pythonpath.extend(v.split(DELIM))
-                        i += 1
-                    # Paths from the core key get appended last, but only
-                    # when home was not set and we haven't found our stdlib
-                    # some other way.
-                    if not home and not stdlib_dir:
-                        v = winreg.QueryValue(key, None)
-                        if isinstance(v, str):
-                            pythonpath.extend(v.split(DELIM))
-                finally:
-                    winreg.CloseKey(key)
-            except OSError:
-                pass
-
     # Then add any entries compiled into the PYTHONPATH macro.
     if PYTHONPATH:
         for p in PYTHONPATH.split(DELIM):
