--- Python-3.11.14/setup.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/setup.py	2026-03-06 22:23:13.985549332 +0400
@@ -971,7 +971,7 @@
         # grp(3)
         self.addext(Extension('grp', ['grpmodule.c']))
 
-        self.addext(Extension('_socket', ['socketmodule.c']))
+        self.addext(Extension('_socket', ['socketmodule.c'], libraries=['resolv', 'socket', 'nsl']))
         self.addext(Extension('spwd', ['spwdmodule.c']))
 
         # select(2); not on ancient System V
