--- Python-3.11.9/configure	2024-04-05 11:13:14.325247469 -0500
+++ Python-3.11.9/configure	2024-04-05 11:14:14.667224175 -0500
@@ -21045,14 +21045,12 @@
 LIBS="-l$LIBREADLINE $READLINE_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
+#include <stdio.h> /* Must be first for Gnu Readline */
+#ifdef WITH_EDITLINE
+# include <editline/readline.h>
+#else
+# include <readline/readline.h>
 #endif
-char rl_pre_input_hook ();
 int
 main ()
 {
@@ -21091,18 +21089,16 @@
 LIBS="-l$LIBREADLINE $READLINE_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
+#include <stdio.h> /* Must be first for Gnu Readline */
+#ifdef WITH_EDITLINE
+# include <editline/readline.h>
+#else
+# include <readline/readline.h>
 #endif
-char rl_completion_display_matches_hook ();
 int
 main ()
 {
-return rl_completion_display_matches_hook ();
+rl_completion_display_matches_hook (NULL, 0, 0);
   ;
   return 0;
 }
