diff --git i/PCbuild/_bsddb.vcproj w/PCbuild/_bsddb.vcproj
index d9886c6..48ba888 100644
--- i/PCbuild/_bsddb.vcproj
+++ w/PCbuild/_bsddb.vcproj
@@ -42,7 +42,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -56,7 +56,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="$(bsddbDepLibs)"
+				AdditionalDependencies="$(bsddbDepLibs) $(prefixDir)\lib"
 				BaseAddress="0x1e180000"
 			/>
 			<Tool
@@ -105,7 +105,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -120,7 +120,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="$(bsddbDepLibs)"
+				AdditionalDependencies="$(bsddbDepLibs) $(prefixDir)\lib"
 				BaseAddress="0x1e180000"
 			/>
 			<Tool
@@ -169,7 +169,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -183,7 +183,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="$(bsddbDepLibs)"
+				AdditionalDependencies="$(prefixDir)\lib\libdb51.lib"
 				BaseAddress="0x1e180000"
 			/>
 			<Tool
@@ -233,7 +233,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -248,7 +248,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="$(bsddbDepLibs)"
+				AdditionalDependencies="libdb51.lib"
 				BaseAddress="0x1e180000"
 			/>
 			<Tool
@@ -297,7 +297,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -361,7 +361,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -425,7 +425,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -489,7 +489,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\.."
+				AdditionalIncludeDirectories="$(bsddbDir),$(bsddbDir)\..,$(prefixDir)\include"
 				PreprocessorDefinitions="_CRT_SECURE_NO_WARNINGS"
 			/>
 			<Tool
@@ -503,7 +503,6 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="$(bsddbDepLibs)"
 				BaseAddress="0x1e180000"
 				TargetMachine="17"
 			/>
@@ -785,7 +784,7 @@
 				>
 			</File>
 			<File
-				RelativePath="$(bsddbDir)\..\dbm\dbm.c"
+				RelativePath="$(bsddbDir)\..\..\lang\dbm\dbm.c"
 				>
 			</File>
 			<File
@@ -941,7 +940,7 @@
 				>
 			</File>
 			<File
-				RelativePath="$(bsddbDir)\..\hsearch\hsearch.c"
+				RelativePath="$(bsddbDir)\..\..\lang\hsearch\hsearch.c"
 				>
 			</File>
 			<File
@@ -1441,18 +1440,6 @@
 				>
 			</File>
 			<File
-				RelativePath="$(bsddbDir)\..\xa\xa.c"
-				>
-			</File>
-			<File
-				RelativePath="$(bsddbDir)\..\xa\xa_db.c"
-				>
-			</File>
-			<File
-				RelativePath="$(bsddbDir)\..\xa\xa_map.c"
-				>
-			</File>
-			<File
 				RelativePath="$(bsddbDir)\..\common\zerofill.c"
 				>
 			</File>
diff --git i/PCbuild/_hashlib.vcproj w/PCbuild/_hashlib.vcproj
index 50d40d1..d78f965 100644
--- i/PCbuild/_hashlib.vcproj
+++ w/PCbuild/_hashlib.vcproj
@@ -153,7 +153,7 @@
 			>
 			<Tool
 				Name="VCPreBuildEventTool"
-				CommandLine="cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;$(PythonExe)&quot; build_ssl.py Release $(PlatformName) -a&#x0D;&#x0A;"
+				CommandLine=""
 			/>
 			<Tool
 				Name="VCCustomBuildTool"
@@ -169,7 +169,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(opensslDir)\inc32"
+				AdditionalIncludeDirectories="$(prefix)\include"
 			/>
 			<Tool
 				Name="VCManagedResourceCompilerTool"
@@ -183,7 +183,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out32\libeay32.lib $(opensslDir)\out32\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefix)\lib\libeay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -216,7 +216,7 @@
 			>
 			<Tool
 				Name="VCPreBuildEventTool"
-				CommandLine="cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;$(PythonExe)&quot; build_ssl.py Release $(PlatformName) -a&#x0D;&#x0A;"
+				CommandLine=""
 			/>
 			<Tool
 				Name="VCCustomBuildTool"
@@ -247,7 +247,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out64\libeay32.lib $(opensslDir)\out64\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefix)\lib\libeay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
diff --git i/PCbuild/_ssl.vcproj w/PCbuild/_ssl.vcproj
index 769be93..328c054 100644
--- i/PCbuild/_ssl.vcproj
+++ w/PCbuild/_ssl.vcproj
@@ -27,7 +27,7 @@
 			>
 			<Tool
 				Name="VCPreBuildEventTool"
-				CommandLine="cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;$(PythonExe)&quot; build_ssl.py Release $(PlatformName) -a&#x0D;&#x0A;"
+				CommandLine=""
 			/>
 			<Tool
 				Name="VCCustomBuildTool"
@@ -57,7 +57,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out32\libeay32.lib $(opensslDir)\out32\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -120,7 +120,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out64\libeay32.lib $(opensslDir)\out64\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -153,7 +153,7 @@
 			>
 			<Tool
 				Name="VCPreBuildEventTool"
-				CommandLine="cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;$(PythonExe)&quot; build_ssl.py Release $(PlatformName) -a&#x0D;&#x0A;"
+				CommandLine=""
 			/>
 			<Tool
 				Name="VCCustomBuildTool"
@@ -169,7 +169,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(opensslDir)\inc32"
+				AdditionalIncludeDirectories="$(prefix)\include"
 			/>
 			<Tool
 				Name="VCManagedResourceCompilerTool"
@@ -183,7 +183,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out32\libeay32.lib $(opensslDir)\out32\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -216,7 +216,7 @@
 			>
 			<Tool
 				Name="VCPreBuildEventTool"
-				CommandLine="cd &quot;$(SolutionDir)&quot;&#x0D;&#x0A;&quot;$(PythonExe)&quot; build_ssl.py Release $(PlatformName) -a&#x0D;&#x0A;"
+				CommandLine=""
 			/>
 			<Tool
 				Name="VCCustomBuildTool"
@@ -233,7 +233,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(opensslDir)\inc64"
+				AdditionalIncludeDirectories="$(prefix)\include"
 			/>
 			<Tool
 				Name="VCManagedResourceCompilerTool"
@@ -247,7 +247,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out64\libeay32.lib $(opensslDir)\out64\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -310,7 +310,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out32\libeay32.lib $(opensslDir)\out32\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -374,7 +374,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out64\libeay32.lib $(opensslDir)\out64\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 				TargetMachine="17"
 			/>
 			<Tool
@@ -438,7 +438,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out32\libeay32.lib $(opensslDir)\out32\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -502,7 +502,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ws2_32.lib $(opensslDir)\out64\libeay32.lib $(opensslDir)\out64\ssleay32.lib"
+				AdditionalDependencies="ws2_32.lib $(prefixDir)\lib\libeay32.lib $(prefixDir)\lib\ssleay32.lib"
 				TargetMachine="17"
 			/>
 			<Tool
diff --git i/PCbuild/bz2.vcproj w/PCbuild/bz2.vcproj
index 088fb46..18bfa43 100644
--- i/PCbuild/bz2.vcproj
+++ w/PCbuild/bz2.vcproj
@@ -166,7 +166,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bz2Dir)"
+				AdditionalIncludeDirectories="$(prefix)\include"
 				PreprocessorDefinitions="WIN32;_FILE_OFFSET_BITS=64"
 			/>
 			<Tool
@@ -180,6 +180,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
+				AdditionalDependencies="$(prefix)\lib\libbz2.lib"
 				BaseAddress="0x1D170000"
 			/>
 			<Tool
@@ -229,7 +230,7 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
-				AdditionalIncludeDirectories="$(bz2Dir)"
+				AdditionalIncludeDirectories="$(prefix)\include"
 				PreprocessorDefinitions="WIN32;_FILE_OFFSET_BITS=64"
 			/>
 			<Tool
@@ -243,6 +244,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
+				AdditionalDependencies="$(prefix)\lib\libbz2.lib"
 				BaseAddress="0x1D170000"
 			/>
 			<Tool
@@ -532,7 +534,7 @@
 			</File>
 		</Filter>
 		<Filter
-			Name="bzip2 1.0.6 Header Files"
+			Name="bzip2 1.0.5 Header Files"
 			>
 			<File
 				RelativePath="$(bz2Dir)\bzlib.h"
@@ -544,7 +546,7 @@
 			</File>
 		</Filter>
 		<Filter
-			Name="bzip2 1.0.6 Source Files"
+			Name="bzip2 1.0.5 Source Files"
 			>
 			<File
 				RelativePath="$(bz2Dir)\blocksort.c"
diff --git i/PCbuild/pcbuild.sln w/PCbuild/pcbuild.sln
index 725c6ea..5697af3 100644
--- i/PCbuild/pcbuild.sln
+++ w/PCbuild/pcbuild.sln
@@ -174,8 +174,8 @@ Global
 		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.PGUpdate|x64.Build.0 = Release|Win32
 		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|Win32.ActiveCfg = Release|Win32
 		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|Win32.Build.0 = Release|Win32
-		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|x64.ActiveCfg = Release|Win32
-		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|x64.Build.0 = Release|Win32
+		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|x64.ActiveCfg = Release|x64
+		{F0E0541E-F17D-430B-97C4-93ADF0DD284E}.Release|x64.Build.0 = Release|x64
 		{CF7AC3D1-E2DF-41D2-BEA6-1E2556CDEA26}.Debug|Win32.ActiveCfg = Debug|Win32
 		{CF7AC3D1-E2DF-41D2-BEA6-1E2556CDEA26}.Debug|Win32.Build.0 = Debug|Win32
 		{CF7AC3D1-E2DF-41D2-BEA6-1E2556CDEA26}.Debug|x64.ActiveCfg = Debug|x64
@@ -238,8 +238,8 @@ Global
 		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.PGUpdate|x64.Build.0 = Release|Win32
 		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|Win32.ActiveCfg = Release|Win32
 		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|Win32.Build.0 = Release|Win32
-		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|x64.ActiveCfg = Release|Win32
-		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|x64.Build.0 = Release|Win32
+		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|x64.ActiveCfg = Release|x64
+		{C73F0EC1-358B-4177-940F-0846AC8B04CD}.Release|x64.Build.0 = Release|x64
 		{28B5D777-DDF2-4B6B-B34F-31D938813856}.Debug|Win32.ActiveCfg = Debug|Win32
 		{28B5D777-DDF2-4B6B-B34F-31D938813856}.Debug|Win32.Build.0 = Debug|Win32
 		{28B5D777-DDF2-4B6B-B34F-31D938813856}.Debug|x64.ActiveCfg = Debug|x64
@@ -269,9 +269,7 @@ Global
 		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.PGUpdate|x64.ActiveCfg = PGUpdate|x64
 		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.PGUpdate|x64.Build.0 = PGUpdate|x64
 		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.Release|Win32.ActiveCfg = Release|Win32
-		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.Release|Win32.Build.0 = Release|Win32
 		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.Release|x64.ActiveCfg = Release|x64
-		{B4D38F3F-68FB-42EC-A45D-E00657BB3627}.Release|x64.Build.0 = Release|x64
 		{0E9791DB-593A-465F-98BC-681011311618}.Debug|Win32.ActiveCfg = Debug|Win32
 		{0E9791DB-593A-465F-98BC-681011311618}.Debug|Win32.Build.0 = Debug|Win32
 		{0E9791DB-593A-465F-98BC-681011311618}.Debug|x64.ActiveCfg = Debug|x64
@@ -413,9 +411,7 @@ Global
 		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.PGUpdate|x64.ActiveCfg = PGUpdate|x64
 		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.PGUpdate|x64.Build.0 = PGUpdate|x64
 		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.Release|Win32.ActiveCfg = Release|Win32
-		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.Release|Win32.Build.0 = Release|Win32
 		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.Release|x64.ActiveCfg = Release|x64
-		{4946ECAC-2E69-4BF8-A90A-F5136F5094DF}.Release|x64.Build.0 = Release|x64
 		{73FCD2BD-F133-46B7-8EC1-144CD82A59D5}.Debug|Win32.ActiveCfg = Debug|Win32
 		{73FCD2BD-F133-46B7-8EC1-144CD82A59D5}.Debug|Win32.Build.0 = Debug|Win32
 		{73FCD2BD-F133-46B7-8EC1-144CD82A59D5}.Debug|x64.ActiveCfg = Debug|x64
diff --git i/PCbuild/pyproject.vsprops w/PCbuild/pyproject.vsprops
index 024bfa8..1f82f7a 100644
--- i/PCbuild/pyproject.vsprops
+++ w/PCbuild/pyproject.vsprops
@@ -45,12 +45,20 @@
 		Value="$(SolutionDir)\python.exe"
 	/>
 	<UserMacro
-		Name="externalsDir"
-		Value="..\.."
+		Name="buildoutDir"
+		Value="..\..\..\..\"
 	/>
 	<UserMacro
+		Name="prefixDir"
+		Value="..\..\..\..\dist\"
+	/>
+	<UserMacro
+		Name="partsDir"
+		Value="..\..\..\"
+   />
+	<UserMacro
 		Name="bsddb47Dir"
-		Value="$(externalsDir)\db-4.7.25.0\build_windows"
+		Value="$(partsDir)\db__compile__\db-5.1.25\src\db"
 	/>
 	<UserMacro
 		Name="bsddb47DepLibs"
@@ -74,15 +81,15 @@
 	/>
 	<UserMacro
 		Name="sqlite3Dir"
-		Value="$(externalsDir)\sqlite-3.6.21"
+		Value="$(partsDir)\sqlite3__compile__\sqlite-autoconf-3070602"
 	/>
 	<UserMacro
 		Name="bz2Dir"
-		Value="$(externalsDir)\bzip2-1.0.6"
+		Value="$(partsDir)\bzip2__compile__\bzip2-1.0.6"
 	/>
 	<UserMacro
 		Name="opensslDir"
-		Value="$(externalsDir)\openssl-1.0.1h"
+		Value="$(partsDir)\openssl__compile__\openssl-1.0.1j"
 	/>
 	<UserMacro
 		Name="tcltkDir"
diff --git i/PCbuild/sqlite3.vcproj w/PCbuild/sqlite3.vcproj
index e1771f4..ef8c328 100644
--- i/PCbuild/sqlite3.vcproj
+++ w/PCbuild/sqlite3.vcproj
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="9,00"
+	Version="9.00"
 	Name="sqlite3"
 	ProjectGUID="{A1A295E5-463C-437F-81CA-1F32367685DA}"
 	RootNamespace="sqlite3"
diff --git i/PCbuild/x64.vsprops w/PCbuild/x64.vsprops
index 9f88d44..2d4dbd6 100644
--- i/PCbuild/x64.vsprops
+++ w/PCbuild/x64.vsprops
@@ -3,7 +3,7 @@
 	ProjectType="Visual C++"
 	Version="8.00"
 	Name="amd64"
-	OutputDirectory="$(SolutionDir)\amd64\"
+	OutputDirectory="$(SolutionDir)"
 	IntermediateDirectory="$(SolutionDir)$(PlatformName)-temp-$(ConfigurationName)\$(ProjectName)\"
 	>
 	<Tool
