diff --git a/Makefile.pre.in b/Makefile.pre.in
index 502317a..b1eddad 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -185,9 +185,6 @@ LIPO_32BIT_FLAGS=@LIPO_32BIT_FLAGS@
 # Options to enable prebinding (for fast startup prior to Mac OS X 10.3)
 OTHER_LIBTOOL_OPT=@OTHER_LIBTOOL_OPT@
 
-# Environment to run shared python without installed libraries
-RUNSHARED=       @RUNSHARED@
-
 # ensurepip options
 ENSUREPIP=      @ENSUREPIP@
 
