--- Python-3.11.14/Makefile.pre.in	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Makefile.pre.in	2026-03-06 22:23:02.458097174 +0400
@@ -1670,7 +1670,7 @@
 ######################################################################
 
 TESTOPTS=	$(EXTRATESTOPTS)
-TESTPYTHON=	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(TESTPYTHONOPTS)
+TESTPYTHON=	$(RUNSHARED) $(BINDIR)/$(PYTHON)$(VERSION) $(TESTPYTHONOPTS)
 TESTRUNNER=	$(TESTPYTHON) $(srcdir)/Tools/scripts/run_tests.py
 TESTTIMEOUT=	1200
 
