--- Python-3.11.16/Modules/_ctypes/stgdict.c	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Modules/_ctypes/stgdict.c	2026-08-12 10:35:44.379955466 +0400
@@ -665,7 +665,7 @@
 /*
  * The value of MAX_STRUCT_SIZE depends on the platform Python is running on.
  */
-#if defined(__aarch64__) || defined(__arm__) || defined(_M_ARM64)
+#if defined(__aarch64__) || defined(__arm__) || defined(_M_ARM64) || defined(__sparc__)
 #  define MAX_STRUCT_SIZE 32
 #elif defined(__powerpc64__)
 #  define MAX_STRUCT_SIZE 64
@@ -891,7 +891,11 @@
                 element_types[element_index++] = &structs[struct_index];
                 structs[struct_index].size = length * edict->ffi_type_pointer.size;
                 structs[struct_index].alignment = edict->ffi_type_pointer.alignment;
+#ifdef FFI_TYPE_NATIVE_ARRAY
+                structs[struct_index].type = FFI_TYPE_NATIVE_ARRAY;
+#else
                 structs[struct_index].type = FFI_TYPE_STRUCT;
+#endif
                 structs[struct_index].elements = &dummy_types[dummy_index];
                 ++struct_index;
                 /* Copy over the element's type, length times. */
