Index: cpython/Lib/distutils/sysconfig.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/Lib/distutils/sysconfig.py	(date 1529223068000)
+++ cpython/Lib/distutils/sysconfig.py	(date 1529223068000)
@@ -426,6 +426,20 @@
     global _config_vars
     _config_vars = {}
     _config_vars.update(build_time_vars)
+    _fix_prefix(_config_vars)
+
+
+def _fix_prefix(config_vars):
+    import sys
+    if config_vars is None:
+        return
+    makefile_prefix = config_vars.get('prefix', None)
+    if not makefile_prefix:
+        return
+    for key, value in config_vars.items():
+        if not isinstance(value, str):
+            continue
+        config_vars[key] = value.replace(makefile_prefix, sys.prefix)
 
 
 def _init_nt():
