Index: cpython/Modules/main.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/Modules/main.c	(date 1529226421000)
+++ cpython/Modules/main.c	(date 1529226421000)
@@ -1727,10 +1727,7 @@
         return _Py_INIT_OK();
     }
 
-    int res = config_get_env_var_dup(&home, L"PYTHONHOME", "PYTHONHOME");
-    if (res < 0) {
-        return DECODE_LOCALE_ERR("PYTHONHOME", res);
-    }
+
     config->home = home;
     return _Py_INIT_OK();
 }
