--- Python-3.11.9/Lib/test/test_socket.py	2025-12-04 21:29:43.410317702 +0000
+++ Python-3.11.9/Lib/test/test_socket.py	2025-12-04 21:34:22.751037584 +0000
@@ -168,7 +168,7 @@
 
 HAVE_SOCKET_VSOCK = _have_socket_vsock()
 
-HAVE_SOCKET_UDPLITE = hasattr(socket, "IPPROTO_UDPLITE")
+HAVE_SOCKET_UDPLITE = hasattr(socket, "IPPROTO_UDPLITE") and not (platform.system() == 'Linux' and 'el6' in platform.release())
 
 HAVE_SOCKET_BLUETOOTH = _have_socket_bluetooth()
 
