diff --git i/Makefile.pre.in w/Makefile.pre.in
index 9d55550..33624be 100644
--- i/Makefile.pre.in
+++ w/Makefile.pre.in
@@ -455,7 +455,7 @@ sharedmods: $(BUILDPYTHON) pybuilddir.txt
 	    *\ -s*|s*) quiet="-q";; \
 	    *) quiet="";; \
 	esac; \
-	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' \
+	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' CONFIG_ARGS='$(CONFIG_ARGS)' OPT='$(OPT)' \
 		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build
 
 # Build static library
