--- Python-3.11.14/Lib/distutils/tests/test_dist.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/distutils/tests/test_dist.py	2026-03-06 22:25:43.902296357 +0400
@@ -444,8 +444,8 @@
         else:
             user_filename = "pydistutils.cfg"
 
-        temp_dir = self.mkdtemp()
-        user_filename = os.path.join(temp_dir, user_filename)
+        user_home = self.mkdtemp()
+        user_filename = os.path.join(user_home, user_filename)
         f = open(user_filename, 'w')
         try:
             f.write('.')
@@ -457,14 +457,16 @@
 
             # linux-style
             if sys.platform in ('linux', 'darwin'):
-                os.environ['HOME'] = temp_dir
+                os.environ['HOME'] = user_home
                 files = dist.find_config_files()
                 self.assertIn(user_filename, files)
 
             # win32-style
             if sys.platform == 'win32':
-                # home drive should be found
-                os.environ['USERPROFILE'] = temp_dir
+                # home directory should be found
+                user_profile = self.mkdtemp()
+                os.environ['HOME'] = user_home
+                os.environ['USERPROFILE'] = user_profile
                 files = dist.find_config_files()
                 self.assertIn(user_filename, files,
                               '%r not found in %r' % (user_filename, files))
--- Python-3.11.14/Lib/ntpath.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/ntpath.py	2026-03-06 22:25:43.903182222 +0400
@@ -332,7 +332,9 @@
     while i < n and path[i] not in _get_bothseps(path):
         i += 1
 
-    if 'USERPROFILE' in os.environ:
+    if 'HOME' in os.environ:
+        userhome = os.environ['HOME']
+    elif 'USERPROFILE' in os.environ:
         userhome = os.environ['USERPROFILE']
     elif not 'HOMEPATH' in os.environ:
         return path
--- Python-3.11.14/Lib/test/test_ntpath.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_ntpath.py	2026-03-06 22:25:43.904646205 +0400
@@ -827,45 +827,48 @@
             env.clear()
             tester('ntpath.expanduser("~test")', '~test')
 
+            env['HOME'] = 'E:\\Home\\eric'
             env['HOMEDRIVE'] = 'C:\\'
             env['HOMEPATH'] = 'Users\\eric'
             env['USERNAME'] = 'eric'
-            tester('ntpath.expanduser("~test")', 'C:\\Users\\test')
-            tester('ntpath.expanduser("~")', 'C:\\Users\\eric')
+            tester('ntpath.expanduser("~test")', 'E:\\Home\\test')
+            tester('ntpath.expanduser("~")', 'E:\\Home\\eric')
 
             del env['HOMEDRIVE']
-            tester('ntpath.expanduser("~test")', 'Users\\test')
-            tester('ntpath.expanduser("~")', 'Users\\eric')
+            tester('ntpath.expanduser("~test")', 'E:\\Home\\test')
+            tester('ntpath.expanduser("~")', 'E:\\Home\\eric')
 
             env.clear()
+            env['HOME'] = 'F:\\Home\\eric'
             env['USERPROFILE'] = 'C:\\Users\\eric'
             env['USERNAME'] = 'eric'
-            tester('ntpath.expanduser("~test")', 'C:\\Users\\test')
-            tester('ntpath.expanduser("~")', 'C:\\Users\\eric')
+            tester('ntpath.expanduser("~test")', 'F:\\Home\\test')
+            tester('ntpath.expanduser("~")', 'F:\\Home\\eric')
             tester('ntpath.expanduser("~test\\foo\\bar")',
-                   'C:\\Users\\test\\foo\\bar')
+                   'F:\\Home\\test\\foo\\bar')
             tester('ntpath.expanduser("~test/foo/bar")',
-                   'C:\\Users\\test/foo/bar')
+                   'F:\\Home\\test/foo/bar')
             tester('ntpath.expanduser("~\\foo\\bar")',
-                   'C:\\Users\\eric\\foo\\bar')
+                   'F:\\Home\\eric\\foo\\bar')
             tester('ntpath.expanduser("~/foo/bar")',
-                   'C:\\Users\\eric/foo/bar')
+                   'F:\\Home\\eric/foo/bar')
 
-            # bpo-36264: ignore `HOME` when set on windows
+            # bpo-36264: do not ignore `HOME` when set on windows
             env.clear()
-            env['HOME'] = 'F:\\'
+            env['HOME'] = 'F:\\Home\\eric'
             env['USERPROFILE'] = 'C:\\Users\\eric'
             env['USERNAME'] = 'eric'
-            tester('ntpath.expanduser("~test")', 'C:\\Users\\test')
-            tester('ntpath.expanduser("~")', 'C:\\Users\\eric')
+            tester('ntpath.expanduser("~test")', 'F:\\Home\\test')
+            tester('ntpath.expanduser("~")', 'F:\\Home\\eric')
 
             # bpo-39899: don't guess another user's home directory if
             # `%USERNAME% != basename(%USERPROFILE%)`
             env.clear()
+            env['HOME'] = 'E:\\Home\\eric'
             env['USERPROFILE'] = 'C:\\Users\\eric'
             env['USERNAME'] = 'idle'
             tester('ntpath.expanduser("~test")', '~test')
-            tester('ntpath.expanduser("~")', 'C:\\Users\\eric')
+            tester('ntpath.expanduser("~")', 'E:\\Home\\eric')
 
 
 
--- Python-3.11.14/Lib/test/test_pathlib.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_pathlib.py	2026-03-06 22:25:43.906404477 +0400
@@ -1506,9 +1506,6 @@
             env['USERPROFILE'] = os.path.join(BASE, 'userprofile')
             self._test_home(self.cls.home())
 
-            # bpo-38883: ignore `HOME` when set on windows
-            env['HOME'] = os.path.join(BASE, 'home')
-            self._test_home(self.cls.home())
 
     def test_samefile(self):
         fileA_path = os.path.join(BASE, 'fileA')
@@ -2808,9 +2805,6 @@
             env['USERPROFILE'] = 'C:\\Users\\alice'
             check()
 
-            # bpo-38883: ignore `HOME` when set on windows
-            env['HOME'] = 'C:\\Users\\eve'
-            check()
 
 
 class CompatiblePathTest(unittest.TestCase):
