--- Python-3.11.16/Lib/test/test_importlib/test_threaded_import.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_importlib/test_threaded_import.py	2026-03-11 00:12:46.103630273 +0400
@@ -254,6 +254,7 @@
                           'partial', 'cfimport.py')
         script_helper.assert_python_ok(fn)
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue43665')
     def test_multiprocessing_pool_circular_import(self):
         # Regression test for bpo-41567
         fn = os.path.join(os.path.dirname(__file__),
