--- Python-3.11.15/Lib/test/test_pty.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/test/test_pty.py	2026-08-13 14:25:01.938793050 +0400
@@ -107,6 +107,7 @@
     def handle_sighup(signum, frame):
         pass
 
+    @unittest.skipIf(sys.platform == 'sunos5', 'not compatible with Solaris pty')
     @expectedFailureIfStdinIsTTY
     def test_openpty(self):
         try:
@@ -191,6 +192,7 @@
         s2 = _readline(master_fd)
         self.assertEqual(b'For my pet fish, Eric.\n', normalize_output(s2))
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue45200')
     def test_fork(self):
         debug("calling pty.fork()")
         pid, master_fd = pty.fork()
@@ -292,6 +294,8 @@
 
         self.assertEqual(data, b"")
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue45223')
+    @unittest.skipIf(sys.platform == 'sunos5', 'not compatible with Solaris pty')
     def test_spawn_doesnt_hang(self):
         self.addCleanup(unlink, TESTFN)
         with open(TESTFN, 'wb') as f:
