diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index b9e2faf..8385ac4 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -418,10 +418,12 @@ def _generate_posix_vars():
 def _init_posix(vars):
     """Initialize the module as appropriate for POSIX systems."""
     # _sysconfigdata is generated at build time, see _generate_posix_vars()
+    from distutils.sysconfig import _fix_prefix
     name = _get_sysconfigdata_name()
     _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
     build_time_vars = _temp.build_time_vars
     vars.update(build_time_vars)
+    _fix_prefix(vars)
 
 def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
