--- Python-3.11.14/Lib/test/test_socket.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_socket.py	2026-03-06 22:24:42.010720063 +0400
@@ -169,7 +169,7 @@
 
 HAVE_SOCKET_VSOCK = _have_socket_vsock()
 
-HAVE_SOCKET_UDPLITE = hasattr(socket, "IPPROTO_UDPLITE")
+HAVE_SOCKET_UDPLITE = hasattr(socket, "IPPROTO_UDPLITE") and not (platform.system() == 'Linux' and 'el6' in platform.release())
 
 HAVE_SOCKET_BLUETOOTH = _have_socket_bluetooth()
 
