diff --git i/Python/pythonrun.c w/Python/pythonrun.c
index e9947e9..6bbb416 100644
--- i/Python/pythonrun.c
+++ w/Python/pythonrun.c
@@ -899,16 +899,7 @@ wchar_t *
 Py_GetPythonHome(void)
 {
     wchar_t *home = default_home;
-    if (home == NULL && !Py_IgnoreEnvironmentFlag) {
-        char* chome = Py_GETENV("PYTHONHOME");
-        if (chome) {
-            size_t size = Py_ARRAY_LENGTH(env_home);
-            size_t r = mbstowcs(env_home, chome, size);
-            if (r != (size_t)-1 && r < size)
-                home = env_home;
-        }
-
-    }
+    // PYTHONHOME is evil, we do not use it
     return home;
 }
 
