--- Python-3.11.15/Lib/test/test_support.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.15/Lib/test/test_support.py	2026-08-11 17:52:52.052153092 +0400
@@ -2,6 +2,7 @@
 import importlib
 import io
 import os
+import platform
 import shutil
 import socket
 import stat
@@ -579,7 +580,9 @@
                                  'Warning -- a\nWarning -- b\n')
 
     def test_has_strftime_extensions(self):
-        if support.is_emscripten or sys.platform == "win32":
+        if (support.is_emscripten
+                or sys.platform == "win32"
+                or (platform.system() == "SunOS" and platform.release() == "5.10")):
             self.assertFalse(support.has_strftime_extensions)
         else:
             self.assertTrue(support.has_strftime_extensions)
