--- Python-3.11.16/Lib/test/test_locale.py	2026-03-03 04:52:57.000000000 +0400
+++ Python-3.11.16/Lib/test/test_locale.py	2026-08-11 17:26:51.143105398 +0400
@@ -3,6 +3,7 @@
 from test.support.warnings_helper import check_warnings
 import unittest
 import locale
+import os
 import sys
 import codecs
 
@@ -367,6 +368,7 @@
         if enc not in ('utf-8', 'iso8859-1', 'cp1252'):
             raise unittest.SkipTest('encoding not suitable')
         if enc != 'iso8859-1' and (sys.platform == 'darwin' or is_android or
+                                  (sys.platform == 'sunos5' and os.uname().release == '5.10') or
                                    sys.platform.startswith('freebsd')):
             raise unittest.SkipTest('wcscoll/wcsxfrm have known bugs')
         BaseLocalizedTest.setUp(self)
