Index: cpython/Modules/socketmodule.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cpython/Modules/socketmodule.c	(date 1532970391000)
+++ cpython/Modules/socketmodule.c	(date 1532970391000)
@@ -5407,10 +5407,8 @@
     Py_buffer buf;
     int res, flag = 0;
 
-#ifdef _AIX
 /* issue #18259, not declared in any useful header file */
 extern int sethostname(const char *, size_t);
-#endif
 
     if (!PyArg_ParseTuple(args, "S:sethostname", &hnobj)) {
         PyErr_Clear();
