--- Python-3.11.14/Lib/_aix_support.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/_aix_support.py	2026-03-06 22:17:47.903042060 +0400
@@ -1,15 +1,9 @@
 """Shared AIX support functions."""
 
+import os
 import sys
 import sysconfig
 
-try:
-    import subprocess
-except ImportError:  # pragma: no cover
-    # _aix_support is used in distutils by setup.py to build C extensions,
-    # before subprocess dependencies like _posixsubprocess are available.
-    import _bootsubprocess as subprocess
-
 
 def _aix_tag(vrtl, bd):
     # type: (List[int], int) -> str
@@ -35,12 +29,10 @@
     builddate reflect the current ABI levels of the runtime environment.
     If no builddate is found give a value that will satisfy pep425 related queries
     """
-    # All AIX systems to have lslpp installed in this location
-    out = subprocess.check_output(["/usr/bin/lslpp", "-Lqc", "bos.rte"])
-    out = out.decode("utf-8")
-    out = out.strip().split(":")  # type: ignore
-    _bd = int(out[-1]) if out[-1] != '' else 9988
-    return (str(out[2]), _bd)
+    uname = os.uname()
+    _rv = '%s.%s.%s.%s' % (uname.version, uname.release, 0, 0)
+    _bd = 9988
+    return (_rv, _bd)
 
 
 def aix_platform():
--- Python-3.11.14/setup.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/setup.py	2026-03-06 22:17:47.901732781 +0400
@@ -1476,6 +1476,7 @@
             Extension(
                 '_decimal',
                 ['_decimal/_decimal.c'],
+                libraries=['m'],
                 # Uncomment for extra functionality:
                 # define_macros=[('EXTRA_FUNCTIONALITY', 1)]
             )
