diff --git a/Makefile.pre.in b/Makefile.pre.in
index f36c11d..454fc70 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -572,7 +572,7 @@ $(LIBRARY): $(LIBRARY_OBJS)
 libpython$(LDVERSION).so: $(LIBRARY_OBJS)
 	if test $(INSTSONAME) != $(LDLIBRARY); then \
 		$(BLDSHARED) -Wl,-h$(INSTSONAME) -o $(INSTSONAME) $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
-		$(LN) -f $(INSTSONAME) $@; \
+		$(LN) -fsn $(INSTSONAME) $@; \
 	else \
 		$(BLDSHARED) -o $@ $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 	fi
