--- Python-3.11.14/Lib/test/test_pty.py	2025-10-09 20:16:55.000000000 +0400
+++ Python-3.11.14/Lib/test/test_pty.py	2026-03-06 22:23:57.376369671 +0400
@@ -191,6 +191,7 @@
         s2 = _readline(master_fd)
         self.assertEqual(b'For my pet fish, Eric.\n', normalize_output(s2))
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue45200')
     def test_fork(self):
         debug("calling pty.fork()")
         pid, master_fd = pty.fork()
@@ -292,6 +293,7 @@
 
         self.assertEqual(data, b"")
 
+    @unittest.skipIf(sys.platform == 'aix', 'https://bugs.python.org/issue45223')
     def test_spawn_doesnt_hang(self):
         self.addCleanup(unlink, TESTFN)
         with open(TESTFN, 'wb') as f:
