diff --git a/Modules/_bsddb.c b/Modules/_bsddb.c
index 6ce86b0..5079ca4 100644
--- a/Modules/_bsddb.c
+++ b/Modules/_bsddb.c
@@ -9766,7 +9766,6 @@ PyMODINIT_FUNC  PyInit__bsddb(void)    /* Note the two underscores */
     ADD_INT(d, DB_REP_PERMANENT);
 
 #if (DBVER >= 44)
-    ADD_INT(d, DB_REP_CONF_NOAUTOINIT);
     ADD_INT(d, DB_REP_CONF_DELAYCLIENT);
     ADD_INT(d, DB_REP_CONF_BULK);
     ADD_INT(d, DB_REP_CONF_NOWAIT);
