--- sqlite-src-3510200/test/attach.test.orig	2026-02-04 03:57:45.594391999 -0500
+++ sqlite-src-3510200/test/attach.test	2026-02-04 04:54:11.203745680 -0500
@@ -759,7 +759,7 @@
     ATTACH DATABASE 'no-such-file' AS nosuch;
   }
 } {0 {}}
-if {$tcl_platform(platform) eq "unix"} {
+if {$tcl_platform(platform) eq "unix" && [exec id -u] > 0} {
   do_test attach-6.2 {
     sqlite3 dbx cannot-read
     dbx eval {CREATE TABLE t1(a,b,c)}
