diff --git a/win/tclWinTime.c b/win/tclWinTime.c
index 0163723..32f3361 100644
--- a/win/tclWinTime.c
+++ b/win/tclWinTime.c
@@ -221,7 +221,7 @@ TclpGetTimeZone(
     int timeZone;
 
     tzset();
-    timeZone = timezone / 60;
+    timeZone = _timezone / 60;
 
     return timeZone;
 }
@@ -658,7 +658,7 @@ TclpGetDate(
 	    return TclpLocaltime(t);
 	}
 
-	time = *t - timezone;
+	time = *t - _timezone;
 
 	/*
 	 * If we aren't near to overflowing the long, just add the bias and
@@ -678,7 +678,7 @@ TclpGetDate(
 	     * Propagate seconds overflow into minutes, hours and days.
 	     */
 
-	    time = tmPtr->tm_sec - timezone;
+	    time = tmPtr->tm_sec - _timezone;
 	    tmPtr->tm_sec = (int)(time % 60);
 	    if (tmPtr->tm_sec < 0) {
 		tmPtr->tm_sec += 60;
