Index: zeromq-4.2.5/src/tweetnacl.c
IDEA additional info: https://github.com/zeromq/libzmq/pull/3037
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- zeromq-4.2.5/src/tweetnacl.c	(date 1534169788000)
+++ zeromq-4.2.5/src/tweetnacl.c	(date 1534169788000)
@@ -46,7 +46,7 @@
 #pragma warning(disable : 4018 4244 4146)
 #endif
 
-// clang-format off
+/* clang-format off */
 
 #include "tweetnacl.h"
 
@@ -926,8 +926,8 @@
 {
     int i;
 #ifndef ZMQ_HAVE_GETRANDOM
-    //  Require that random_open has already been called, to avoid
-    //  race conditions.
+    /*  Require that random_open has already been called, to avoid */
+    /*  race conditions. */
     assert (fd != -1);
 #endif
     while (xlen > 0) {
@@ -950,7 +950,7 @@
     }
 }
 
-//  Do not call manually! Use random_close from random.hpp
+/*  Do not call manually! Use random_close from random.hpp */
 int randombytes_close (void)
 {
     int rc = -1;
@@ -959,11 +959,11 @@
         fd = -1;
         rc = 0;
     }
-#endif // ZMQ_HAVE_GETRANDOM
+#endif /* ZMQ_HAVE_GETRANDOM */
     return rc;
 }
 
-//  Do not call manually! Use random_open from random.hpp
+/*  Do not call manually! Use random_open from random.hpp */
 int sodium_init (void)
 {
 #ifndef ZMQ_HAVE_GETRANDOM
@@ -983,11 +983,11 @@
         assert (rc != -1);
 #endif
     }
-#endif // ZMQ_HAVE_GETRANDOM
+#endif /* ZMQ_HAVE_GETRANDOM */
     return 0;
 }
 
 #endif
 
 #endif
-// clang-format on
+/* clang-format on */
