diff --git i/win32/Makefile.msc w/win32/Makefile.msc
index a731c0c..d374ba4 100644
--- i/win32/Makefile.msc
+++ w/win32/Makefile.msc
@@ -21,7 +21,7 @@ AS = ml
 LD = link
 AR = lib
 RC = rc
-CFLAGS  = -nologo -MD -W3 -O2 -Oy- -Zi -Fd"zlib" $(LOC)
+CFLAGS  = -nologo -MT -W3 -O2 -Oy- -Zi -Fd"zlib" $(LOC)
 WFLAGS  = -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE
 ASFLAGS = -coff -Zi
 LDFLAGS = -nologo -debug -incremental:no -opt:ref
@@ -137,3 +137,11 @@ clean:
 	-del *.pdb
 	-del *.manifest
 	-del foo.gz
+
+install:
+    -cp zlib.h $(PREFIX)\include
+    -cp zlibdefs.h $(PREFIX)\include
+    -cp zconf.h $(PREFIX)\include
+    -cp *.lib $(PREFIX)\lib
+    -cp *.dll $(PREFIX)\lib
+
