diff -ur zstd-1.5.6/Makefile zstd-1.5.6/Makefile
--- zstd-1.5.6/Makefile	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/Makefile	2025-12-05 13:40:29.328650615 +0000
@@ -151,7 +151,7 @@
 #------------------------------------------------------------------------------
 # make install is validated only for Linux, macOS, Hurd and some BSD targets
 #------------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT CYGWIN_NT Haiku AIX))
+ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT CYGWIN_NT Haiku SunOS AIX))
 
 HOST_OS = POSIX
 
diff -ur zstd-1.5.6/lib/Makefile zstd-1.5.6/lib/Makefile
--- zstd-1.5.6/lib/Makefile	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/lib/Makefile	2025-12-05 13:40:29.328650615 +0000
@@ -140,7 +140,7 @@
 LIBZSTD = libzstd.$(SHARED_EXT_VER)
 .PHONY: $(LIBZSTD)  # must be run every time
 $(LIBZSTD): CPPFLAGS += $(CPPFLAGS_DYNLIB)
-$(LIBZSTD): CFLAGS   += -fPIC -fvisibility=hidden
+$(LIBZSTD): CFLAGS   += -fPIC
 $(LIBZSTD): LDFLAGS  += -shared $(LDFLAGS_DYNLIB)
 
 ifndef BUILD_DIR
@@ -239,7 +239,7 @@
 # but they will correctly behave single-threaded.
 ZSTDMT_FILES = zstdmt_compress.c
 ZSTD_NOMT_FILES = $(filter-out $(ZSTDMT_FILES),$(notdir $(ZSTD_FILES)))
-libzstd-nomt: CFLAGS += -fPIC -fvisibility=hidden
+libzstd-nomt: CFLAGS += -fPIC
 libzstd-nomt: LDFLAGS += -shared
 libzstd-nomt: $(ZSTD_NOMT_FILES)
 	@echo compiling single-thread dynamic library $(LIBVER)
@@ -320,7 +320,7 @@
 	        $< >$@
 
 .PHONY: install
-install: install-pc install-static install-shared install-includes
+install: install-pc install-shared install-includes
 	@echo zstd static and shared library installed
 
 .PHONY: install-pc
diff -ur zstd-1.5.6/lib/common/mem.h zstd-1.5.6/lib/common/mem.h
--- zstd-1.5.6/lib/common/mem.h	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/lib/common/mem.h	2025-12-05 13:40:29.328650615 +0000
@@ -133,7 +133,9 @@
  */
 #ifndef MEM_FORCE_MEMORY_ACCESS   /* can be defined externally, on command line for example */
 #  ifdef __GNUC__
-#    define MEM_FORCE_MEMORY_ACCESS 1
+#    ifndef __sparc__
+#      define MEM_FORCE_MEMORY_ACCESS 1
+#    endif
 #  endif
 #endif
 
diff -ur zstd-1.5.6/programs/Makefile zstd-1.5.6/programs/Makefile
--- zstd-1.5.6/programs/Makefile	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/programs/Makefile	2025-12-05 13:40:53.161883214 +0000
@@ -27,6 +27,10 @@
   ALIGN_LOOP =
 endif
 
+ifeq ($(shell uname -r | grep -c el6), 1)
+  LDFLAGS += -lrt
+endif
+
 ZSTDLIB_COMMON_SRC := $(sort $(ZSTD_COMMON_FILES))
 ZSTDLIB_COMPRESS_SRC := $(sort $(ZSTD_COMPRESS_FILES))
 ZSTDLIB_DECOMPRESS_SRC := $(sort $(ZSTD_DECOMPRESS_FILES))
diff -ur zstd-1.5.6/programs/platform.h zstd-1.5.6/programs/platform.h
--- zstd-1.5.6/programs/platform.h	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/programs/platform.h	2025-12-05 13:40:29.328650615 +0000
@@ -59,8 +59,8 @@
 #  if !defined(_LARGEFILE_SOURCE)                  /* obsolete macro, replaced with _FILE_OFFSET_BITS */
 #    define _LARGEFILE_SOURCE 1                    /* Large File Support extension (LFS) - fseeko, ftello */
 #  endif
-#  if defined(_AIX) || defined(__hpux)
-#    define _LARGE_FILES                           /* Large file support on 32-bits AIX and HP-UX */
+#  if defined(__hpux)
+#    define _LARGE_FILES                           /* Large file support on 32-bits HP-UX */
 #  endif
 #endif
 
diff -ur zstd-1.5.6/tests/playTests.sh zstd-1.5.6/tests/playTests.sh
--- zstd-1.5.6/tests/playTests.sh	2024-03-21 22:05:51.000000000 +0000
+++ zstd-1.5.6/tests/playTests.sh	2025-12-05 13:40:29.329650620 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e # exit immediately on error
 # set -x # print commands before execution (debug)
@@ -94,11 +94,7 @@
 TESTDIR="$SCRIPT_DIR/../tests"
 UNAME=${UNAME:-$(uname)}
 GREP=${GREP:-grep}
-
-case "$UNAME" in
-  SunOS) DIFF=${DIFF:-gdiff} ;;
-  *) DIFF=${DIFF:-diff} ;;
-esac
+DIFF=${DIFF:-diff}
 
 detectedTerminal=false
 if [ -t 0 ] && [ -t 1 ]
@@ -330,7 +326,7 @@
 zstd -d -f tmp_corrupt.zst --check --no-check # final flag overrides
 zstd -d -f tmp.zst --no-check
 
-if [ "$isWindows" = false ] && [ "$UNAME" != "AIX" ]; then
+if [ "$isWindows" = false ] && [ "$UNAME" != "AIX" ] && [ "$UNAME" != "SunOS" ]; then
   if [ -n "$(which readelf)" ]; then
     println "test: check if binary has executable stack (#2963)"
     readelf -lW "$ZSTD_BIN" | $GREP 'GNU_STACK .* RW ' || die "zstd binary has executable stack!"
