diff -ur zstd-1.5.5/tests/playTests.sh zstd-1.5.5/tests/playTests.sh
--- zstd-1.5.5/tests/playTests.sh	2023-11-23 06:14:24.589076505 -0600
+++ zstd-1.5.5/tests/playTests.sh	2023-11-23 06:18:04.130293007 -0600
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 unset ZSTD_CLEVEL
 unset ZSTD_NBTHREADS
@@ -80,6 +80,14 @@
     printf '%b\n' "${*}"
 }
 
+get_mtime() {
+    python3 -c 'import os, sys; print(int(os.stat(sys.argv[1]).st_mtime))' "$1"
+}
+
+get_perms() {
+    python3 -c 'import os, stat, sys; print("%o" % stat.S_IMODE(os.stat(sys.argv[1]).st_mode))' "$1"
+}
+
 if [ -z "${size}" ]; then
     size=
 else
@@ -117,6 +125,7 @@
   FreeBSD) MD5SUM="gmd5sum" ;;
   NetBSD) MD5SUM="md5 -n" ;;
   OpenBSD) MD5SUM="md5" ;;
+  AIX) MD5SUM="openssl md5" ;;
   *) MD5SUM="md5sum" ;;
 esac
 
@@ -123,6 +132,7 @@
 MTIME="stat -c %Y"
 case "$UNAME" in
     Darwin | FreeBSD | OpenBSD | NetBSD) MTIME="stat -f %m" ;;
+    AIX) MTIME=get_mtime ;;
 esac
 
 assertSameMTime() {
@@ -135,6 +145,7 @@
 GET_PERMS="stat -c %a"
 case "$UNAME" in
     Darwin | FreeBSD | OpenBSD | NetBSD) GET_PERMS="stat -f %Lp" ;;
+    AIX) GET_PERMS=get_perms ;;
 esac
 
 assertFilePermissions() {
diff -ur zstd-1.5.5/tests/cli-tests/common/mtime.sh zstd-1.5.5/tests/cli-tests/common/mtime.sh
--- zstd-1.5.5/tests/cli-tests/common/mtime.sh	2023-04-04 15:13:52.000000000 -0500
+++ zstd-1.5.5/tests/cli-tests/common/mtime.sh	2023-11-23 06:19:15.585222265 -0600
@@ -1,8 +1,13 @@
 . "$COMMON/platform.sh"
 
+get_mtime() {
+    python3 -c 'import os, sys; print(int(os.stat(sys.argv[1]).st_mtime))' "$1"
+}
+
 MTIME="stat -c %Y"
 case "$UNAME" in
     Darwin | FreeBSD | OpenBSD | NetBSD) MTIME="stat -f %m" ;;
+    AIX) MTIME=get_mtime ;;
 esac
 
 assertSameMTime() {
diff -ur zstd-1.5.5/tests/cli-tests/common/permissions.sh zstd-1.5.5/tests/cli-tests/common/permissions.sh
--- zstd-1.5.5/tests/cli-tests/common/permissions.sh	2023-04-04 15:13:52.000000000 -0500
+++ zstd-1.5.5/tests/cli-tests/common/permissions.sh	2023-11-23 06:19:59.307762916 -0600
@@ -1,8 +1,13 @@
 . "$COMMON/platform.sh"
 
+get_perms() {
+    python3 -c 'import os, stat, sys; print("%o" % stat.S_IMODE(os.stat(sys.argv[1]).st_mode))' "$1"
+}
+
 GET_PERMS="stat -c %a"
 case "$UNAME" in
     Darwin | FreeBSD | OpenBSD | NetBSD) GET_PERMS="stat -f %Lp" ;;
+    AIX) GET_PERMS=get_perms ;;
 esac
 
 assertFilePermissions() {
